<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Ms.GSI">
<title>*Ms.GSI* in the multistage of genetic stock identification madness • Ms.GSI</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="*Ms.GSI* in the multistage of genetic stock identification madness">
<meta property="og:description" content="Ms.GSI">
<meta property="og:image" content="https://boppingshoe.github.io/Ms.GSI/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Ms.GSI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/msgsi_vignette.html">*Ms.GSI* in the multistage of genetic stock identification madness</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/boppingshoe/Ms.GSI/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>*Ms.GSI* in the multistage of genetic stock identification madness</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/boppingshoe/Ms.GSI/blob/HEAD/vignettes/msgsi_vignette.Rmd" class="external-link"><code>vignettes/msgsi_vignette.Rmd</code></a></small>
      <div class="d-none name"><code>msgsi_vignette.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boppingshoe/Ms.GSI" class="external-link">Ms.GSI</a></span><span class="op">)</span></span>
<span><span class="co"># devtools::load_all()</span></span></code></pre></div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>This document contains the background information for a integrated
multistage genetic stock identification (GSI) model in two parts. The
first part the describes how to use package <em>Ms.GSI</em> to conduct a
GSI analysis. The steps include formatting input data, running the
integrated multistage GSI model, and summarizing results. The second
part details the technical background of integrated multistage GSI model
and its mathematical theory. There is a separate article describe the
general background of the integrated multistage framework (<em>add link
to integrated multistage paper</em>).</p>
</div>
<div class="section level2">
<h2 id="how-to-use-ms-gsi">How to use <em>Ms.GSI</em><a class="anchor" aria-label="anchor" href="#how-to-use-ms-gsi"></a>
</h2>
<p><em>Ms.GSI</em> follows the work flow: format input data, run
integrated multistage model, and summarize results/convergence
diagnostics.</p>
<div class="section level3">
<h3 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h3>
<p>There are few pieces of information needed for input data set:</p>
<ul>
<li>broad-scale baseline</li>
<li>regional baseline</li>
<li>mixture sample</li>
<li>broad-scale population information</li>
<li>regional population information</li>
</ul>
<p>There are pre-loaded example data sets available in <em>Ms.GSI</em>.
We will look at them one at a time. Note that the example data sets are
simulated using existing baseline archived by the Alaska Department of
Fish &amp; Game Gene Conservation Lab (GCL). The fabricated data set
does not represent the true population proportions in real fisheries. In
this example, we made up a scenario similar to the Bering Sea groundfish
fisheries where Chinook salmon harvested as bycatch originated from a
wide range of geographic areas. Within the bycatch sample, we were
interested in the proportion contribution from the lower, middle and
upper Yukon River reporting groups (see map below). We used a coast-wide
data set for Chinook salmon (Templin et al. 2011; Templin baseline
hereafter) as the broad-scale baseline to separate Yukon River fish from
other non-Yukon stocks in our data set during the first stage of the
analysis. However, genetic markers in the Templin baseline were unable
to clearly distinguish between lower Yukon River and other coastal
western Alaska populations, so we used a second baseline with additional
genetic markers that were specifically designed to differentiate Yukon
River Chinook salmon populations (Lee et al. 2021; Yukon River baseline
hereafter) as the regional fine-scale baseline for the second stage.</p>
<p>It is important to note that in the original grouping of the Templin
baseline, Lower Yukon was a part of Coastal Western Alaska reporting
group. We isolated Lower Yukon from the rest of the Coastal Western
Alaska, but the accuracy of the proportion estimates would likely
diminish because of the breakup of Coastal Western Alaska reporting
group. We do not recommend using a genetic baseline beyond its original
design. And researchers should be aware of the capability of their
genetic baselines before utilizing them in the integrated multistage
model. In our example, it would be ideal to keep the Coastal Western
Alaska group intact in the broad-scale baseline, and break up the group
into Lower Yukon and others using a fine-scale regional baseline.
However, at the time of writing, such regional baseline with adequate
resolution was still in development.</p>
<div class="figure">
<img src="small_map_bering_chinook.png" style="width:100.0%" alt=""><p class="caption">Collection locations and color-coded reporting groups
of Chinook salmon represented in the Yukon River example. Shaded area
represents the Yukon River region.</p>
</div>
<p>We assembled a mixture sample containing 150 individuals from
collection sites across Yukon River, coastal western Alaskan, Alaska
Peninsula, Gulf of Alaska, and Kamchatka Peninsula (Russia). The
collections were grouped into five reporting groups: Lower Yukon, Middle
Yukon, Upper Yukon, Coastal Western Alaska (Coastal West Alaska), and
Others.</p>
<div class="section level4">
<h4 id="mixture">Mixture<a class="anchor" aria-label="anchor" href="#mixture"></a>
</h4>
<p>First we will take a look at the baseline and mixture samples.
<em>Ms.GSI</em> accepts the genotype information in two format: 1) GCL
format or 2) package <em>rubias</em> format. The example data sets are
in <em>rubias</em> format and naming convention, but procedures for GCL
format are the same<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;Note for the GCL folks: you don’t need to transform GCL
data sets to &lt;em&gt;rubias&lt;/em&gt; format. &lt;em&gt;Ms.GSI&lt;/em&gt; got you covered!&lt;/p&gt;"><sup>1</sup></a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">mix</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 150 × 358</span></span></span>
<span><span class="co">#&gt;    sample_type repunit collection indiv   `GTH2B-550` `GTH2B-550.1` NOD1  NOD1.1</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_1  C           G             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_2  C           C             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_3  C           C             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_4  C           C             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_5  C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_6  C           C             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_7  C           C             <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_8  C           G             <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_9  C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_10 C           G             <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 140 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 350 more variables: `Ots_100884-287` &lt;chr&gt;, `Ots_100884-287.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_101554-407` &lt;chr&gt;, `Ots_101554-407.1` &lt;chr&gt;, `Ots_102414-395` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102414-395.1` &lt;chr&gt;, `Ots_102867-609` &lt;chr&gt;, `Ots_102867-609.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103041-52` &lt;chr&gt;, `Ots_103041-52.1` &lt;chr&gt;, `Ots_103122-180` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103122-180.1` &lt;chr&gt;, `Ots_104048-194` &lt;chr&gt;, `Ots_104048-194.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_104063-132` &lt;chr&gt;, `Ots_104063-132.1` &lt;chr&gt;, `Ots_104415-88` &lt;chr&gt;, …</span></span></span></code></pre></div>
<p>Columns 5 to 358 contain genotype information for loci in
<strong>BOTH</strong> broad-scale and regional baselines. You do not
have to specify the loci for each baseline (you can if you want to
double check, more on that later). <em>Ms.GSI</em> matches the loci
between mixture and baselines as long as the locus names are
consistent.</p>
<p>If you have fish with known-origin, you can specify their identities
by adding a column called <code>known_collection</code> in the mixture
data set. The entry for known-origin should match the collection name in
the broad-scale baseline. Fish with unknown-origin should have a
<code>NA</code> entry.</p>
</div>
<div class="section level4">
<h4 id="broad-scale-baseline">Broad-scale baseline<a class="anchor" aria-label="anchor" href="#broad-scale-baseline"></a>
</h4>
<p>Next, we will take a look at the broad-scale (Templin) baseline
example provided in <em>Ms.GSI</em>. There are originally 45 loci in the
Templin baseline, but we reduced the marker set to 28 loci due to
limitation on the data size (and other technical reasons). However, for
demonstration purpose, this data set will suffice.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">base_templin</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 29,363 × 60</span></span></span>
<span><span class="co">#&gt;    sample_type repunit collection indiv   `GTH2B-550` `GTH2B-550.1` NOD1  NOD1.1</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> reference   Russia  KBIST98L   KBIST9… C           C             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> reference   Russia  KBIST98L   KBIST9… C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> reference   Russia  KBIST98L   KBIST9… C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> reference   Russia  KBIST98L   KBIST9… C           C             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> reference   Russia  KBIST98L   KBIST9… C           G             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> reference   Russia  KBIST98L   KBIST9… C           C             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> reference   Russia  KBIST98L   KBIST9… C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> reference   Russia  KBIST98L   KBIST9… C           C             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> reference   Russia  KBIST98L   KBIST9… C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> reference   Russia  KBIST98L   KBIST9… C           C             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 29,353 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 52 more variables: `Ots_2KER-137` &lt;chr&gt;, `Ots_2KER-137.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_AsnRS-72` &lt;chr&gt;, `Ots_AsnRS-72.1` &lt;chr&gt;, Ots_ETIF1A &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Ots_ETIF1A.1 &lt;chr&gt;, `Ots_GPH-318` &lt;chr&gt;, `Ots_GPH-318.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_GST-207` &lt;chr&gt;, `Ots_GST-207.1` &lt;chr&gt;, `Ots_hnRNPL-533` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_hnRNPL-533.1` &lt;chr&gt;, `Ots_HSP90B-100` &lt;chr&gt;, `Ots_HSP90B-100.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_IGF1-91` &lt;chr&gt;, `Ots_IGF1-91.1` &lt;chr&gt;, `Ots_IK1-328` &lt;chr&gt;, …</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="regional-baseline">Regional baseline<a class="anchor" aria-label="anchor" href="#regional-baseline"></a>
</h4>
<p>The regional baseline (Yukon) is in the same format. There are
originally 380 loci in the Yukon River Chinook baseline, but we reduced
the numbers to 177 in this demonstration.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">base_yukon</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5,435 × 358</span></span></span>
<span><span class="co">#&gt;    sample_type repunit   collection indiv `GTH2B-550` `GTH2B-550.1` NOD1  NOD1.1</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> reference   Lower Yu… KANDR02.K… KAND… G           G             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> reference   Lower Yu… KANDR02.K… KAND… C           G             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> reference   Lower Yu… KANDR02.K… KAND… G           G             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> reference   Lower Yu… KANDR02.K… KAND… G           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> reference   Lower Yu… KANDR02.K… KAND… C           G             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> reference   Lower Yu… KANDR02.K… KAND… C           C             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> reference   Lower Yu… KANDR02.K… KAND… C           G             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> reference   Lower Yu… KANDR02.K… KAND… C           C             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> reference   Lower Yu… KANDR02.K… KAND… C           C             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> reference   Lower Yu… KANDR02.K… KAND… C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 5,425 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 350 more variables: `Ots_100884-287` &lt;chr&gt;, `Ots_100884-287.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_101554-407` &lt;chr&gt;, `Ots_101554-407.1` &lt;chr&gt;, `Ots_102414-395` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102414-395.1` &lt;chr&gt;, `Ots_102867-609` &lt;chr&gt;, `Ots_102867-609.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103041-52` &lt;chr&gt;, `Ots_103041-52.1` &lt;chr&gt;, `Ots_103122-180` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103122-180.1` &lt;chr&gt;, `Ots_104048-194` &lt;chr&gt;, `Ots_104048-194.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_104063-132` &lt;chr&gt;, `Ots_104063-132.1` &lt;chr&gt;, `Ots_104415-88` &lt;chr&gt;, …</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="population-information">Population information<a class="anchor" aria-label="anchor" href="#population-information"></a>
</h4>
<p>Another piece of information needed is the population details for
each baseline. You need to include three columns in the population
information table. Column <code>collection</code> contains the names for
each population in the baseline. Column <code>reunit</code> specifies
the reporting group each population belongs to. Column
<code>grpvec</code> specifies the identification number for each
reporting group. Below shows the first ten rows of population
information for the Templin (broad-scale) baseline.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">templin_pops211</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 211 × 3</span></span></span>
<span><span class="co">#&gt;    collection       repunit             grpvec</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> KBIST98L         Russia                   1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> KBOLS02.KBOLS98E Russia                   1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> KKAMC97.KKAMC98L Russia                   1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> KPAKH02          Russia                   1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> KPILG05.KPILG06  Coastal West Alaska      2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> KUNAL05          Coastal West Alaska      2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> KGOLS05.KGOLS06  Coastal West Alaska      2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> KANDR02.KANDR03  Lower Yukon              3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> KANVI02          Lower Yukon              3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> KGISA01          Lower Yukon              3</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 201 more rows</span></span></span></code></pre></div>
<p>If you have hatchery populations in your mixture sample, you can tell
<em>Ms.GSI</em> either a collection belongs to natural or
hatchery-origin by adding an <code>origin</code> column in the
population information table. In the <code>origin</code> column, you
identify each collection with <code>"wild"</code> or
<code>"hatchery"</code>. If you don’t care to separate natural and
hatchery origins, you can lump them in one collection. In this case, you
don’t need to add an <code>origin</code> column. Also, this option to
identify hatchery fish is only available in the broad-scale baseline, so
don’t add an <code>origin</code> column in the population table for
regional baseline.</p>
<p>Population information table for the Yukon (regional) baseline is in
the same format, but not necessarily in the same order.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">yukon_pops50</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 50 × 3</span></span></span>
<span><span class="co">#&gt;    collection                          grpvec repunit     </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> KANDR02.KANDR03                          1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> KANVI03.KANVI07                          1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> KNUL12SF                                 1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> KNUL12NF                                 1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> KGISA01                                  1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> KKATE02.KKATE12                          1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> KHENS01                                  2 Middle Yukon</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> KHENS07.KHENS15                          2 Middle Yukon</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> KSFKOY03                                 2 Middle Yukon</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> KMFKOY10.KMFKOY11.KMFKOY12.KMFKOY13      2 Middle Yukon</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 40 more rows</span></span></span></code></pre></div>
<p>Once you have the data files ready (I recommend saving them as .Rdata
or .Rds files), you can use <code><a href="../reference/prep_msgsi_data.html">prep_msgsi_data()</a></code> function to
convert them into input data set for model run. You’ll also need to
identify “groups of interest” in parameter <code>sub-group</code>. In
this example, groups of interests are Lower Yukon, Middle Yukon and
Upper Yukon reporting groups. Their identify numbers are 3, 4, and 5 in
the broad-scale baseline. There’s an option to save the input data at a
designated directory by identifying the location in parameter
<code>file_path</code>.</p>
<p><code><a href="../reference/prep_msgsi_data.html">prep_msgsi_data()</a></code> function matches the loci between
mixture and baselines. But if you want to make sure that you didn’t miss
any locus in your baselines or mixture, you can manually provide loci
names (in string vector) for each baseline by inputting them in
<code>loci1</code> and <code>loci2</code>. In this example we don’t
manually provide lists of loci because we trust that mixture and
baselines all have the correct loci.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">msgsi_dat</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/prep_msgsi_data.html">prep_msgsi_data</a></span><span class="op">(</span>mixture_data <span class="op">=</span> <span class="va">mix</span>,</span>
<span>  baseline1_data <span class="op">=</span> <span class="va">base_templin</span>, baseline2_data <span class="op">=</span> <span class="va">base_yukon</span>,</span>
<span>  pop1_info <span class="op">=</span> <span class="va">templin_pops211</span>, pop2_info <span class="op">=</span> <span class="va">yukon_pops50</span>, sub_group <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Compiling input data, may take a minute or two...</span></span>
<span><span class="co">#&gt; Time difference of 12.85391 secs</span></span></code></pre></div>
<p><code><a href="../reference/prep_msgsi_data.html">prep_msgsi_data()</a></code> formats the data files and put them in
a list. It took around ten seconds to format the input data in this
case. Bigger data sets may take longer. Here are the first few
rows/items in the input data list:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">msgsi_dat</span>, <span class="va">head</span><span class="op">)</span></span>
<span><span class="co">#&gt; $x</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 74</span></span></span>
<span><span class="co">#&gt;   indiv    `GTH2B-550_1` `GTH2B-550_2` NOD1_1 NOD1_2 `Ots_2KER-137_1`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> fish_1               1             1      2      0                0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> fish_10              1             1      0      0                1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> fish_100             1             1      1      1                1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> fish_101             1             1      1      1                1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> fish_102             0             2      0      2                0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> fish_103             2             0      1      1                1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 68 more variables: `Ots_2KER-137_2` &lt;int&gt;, `Ots_2KER-137_3` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_AsnRS-72_1` &lt;int&gt;, `Ots_AsnRS-72_2` &lt;int&gt;, `Ots_AsnRS-72_3` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Ots_ETIF1A_1 &lt;int&gt;, Ots_ETIF1A_2 &lt;int&gt;, `Ots_GPH-318_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_GPH-318_2` &lt;int&gt;, `Ots_GST-207_1` &lt;int&gt;, `Ots_GST-207_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_HSP90B-100_1` &lt;int&gt;, `Ots_HSP90B-100_2` &lt;int&gt;, `Ots_IGF1-91_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_IGF1-91_2` &lt;int&gt;, `Ots_IGF1-91_3` &lt;int&gt;, `Ots_IK1-328_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_IK1-328_2` &lt;int&gt;, `Ots_IK1-328_3` &lt;int&gt;, `Ots_LEI-292_1` &lt;int&gt;, …</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $x2</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 355</span></span></span>
<span><span class="co">#&gt;   indiv    `GTH2B-550_1` `GTH2B-550_2` NOD1_1 NOD1_2 `Ots_100884-287_1`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> fish_1               1             1      2      0                  2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> fish_10              1             1      0      0                  0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> fish_100             1             1      1      1                  1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> fish_101             1             1      1      1                  2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> fish_102             2             0      0      2                  1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> fish_103             0             2      1      1                  1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 349 more variables: `Ots_100884-287_2` &lt;int&gt;, `Ots_101554-407_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_101554-407_2` &lt;int&gt;, `Ots_102414-395_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102414-395_2` &lt;int&gt;, `Ots_102867-609_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102867-609_2` &lt;int&gt;, `Ots_103041-52_1` &lt;int&gt;, `Ots_103041-52_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103122-180_1` &lt;int&gt;, `Ots_103122-180_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_104048-194_1` &lt;int&gt;, `Ots_104048-194_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_104063-132_1` &lt;int&gt;, `Ots_104063-132_2` &lt;int&gt;, …</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $y</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 76</span></span></span>
<span><span class="co">#&gt;   collection            repunit grpvec `GTH2B-550_1` `GTH2B-550_2` NOD1_1 NOD1_2</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CHBIG92.KIBIG93.KBIG… Northe…      9           254            82    104    228</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> CHCRY92.KICRY94.KCRY… Coasta…     10           306           302    120    488</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> CHDMT92.KDEER94       Coasta…     10           178           116     77    217</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> CHKAN92.KIKAN93.KKAN… Coasta…      2           341           147    281    199</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> CHKOG92.KIKOG93.KKOG… Coasta…      2           205            91    191    105</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> CHNUU92.KINUS93       Coasta…      2            85            27     73     41</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 69 more variables: `Ots_2KER-137_1` &lt;int&gt;, `Ots_2KER-137_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_2KER-137_3` &lt;int&gt;, `Ots_AsnRS-72_1` &lt;int&gt;, `Ots_AsnRS-72_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_AsnRS-72_3` &lt;int&gt;, Ots_ETIF1A_1 &lt;int&gt;, Ots_ETIF1A_2 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_GPH-318_1` &lt;int&gt;, `Ots_GPH-318_2` &lt;int&gt;, `Ots_GST-207_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_GST-207_2` &lt;int&gt;, `Ots_HSP90B-100_1` &lt;int&gt;, `Ots_HSP90B-100_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_IGF1-91_1` &lt;int&gt;, `Ots_IGF1-91_2` &lt;int&gt;, `Ots_IGF1-91_3` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_IK1-328_1` &lt;int&gt;, `Ots_IK1-328_2` &lt;int&gt;, `Ots_IK1-328_3` &lt;int&gt;, …</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $y2</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 357</span></span></span>
<span><span class="co">#&gt;   collection            grpvec repunit `GTH2B-550_1` `GTH2B-550_2` NOD1_1 NOD1_2</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CHSID92j                   3 Upper …             7           183    116     74</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> K100MILECR16.K100MIL…      3 Upper …             7           103     78     34</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> KANDR02.KANDR03            1 Lower …            78           230    208    100</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> KANVI03.KANVI07            1 Lower …            62           164    131     79</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> KBEAV97                    2 Middle…            40           148    152     38</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> KBIGS87.KBIGS07            3 Upper …            30           258    231     65</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 350 more variables: `Ots_100884-287_1` &lt;int&gt;, `Ots_100884-287_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_101554-407_1` &lt;int&gt;, `Ots_101554-407_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102414-395_1` &lt;int&gt;, `Ots_102414-395_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102867-609_1` &lt;int&gt;, `Ots_102867-609_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103041-52_1` &lt;int&gt;, `Ots_103041-52_2` &lt;int&gt;, `Ots_103122-180_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103122-180_2` &lt;int&gt;, `Ots_104048-194_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_104048-194_2` &lt;int&gt;, `Ots_104063-132_1` &lt;int&gt;, …</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $iden</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $nalleles</span></span>
<span><span class="co">#&gt;    GTH2B-550         NOD1 Ots_2KER-137 Ots_AsnRS-72   Ots_ETIF1A  Ots_GPH-318 </span></span>
<span><span class="co">#&gt;            2            2            3            3            2            2 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $nalleles2</span></span>
<span><span class="co">#&gt;      GTH2B-550           NOD1 Ots_100884-287 Ots_101554-407 Ots_102414-395 </span></span>
<span><span class="co">#&gt;              2              2              2              2              2 </span></span>
<span><span class="co">#&gt; Ots_102867-609 </span></span>
<span><span class="co">#&gt;              2 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $groups</span></span>
<span><span class="co">#&gt; [1]  9 10 10  2  2  2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $p2_groups</span></span>
<span><span class="co">#&gt; [1] 3 3 1 1 2 3</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $sub_group</span></span>
<span><span class="co">#&gt; [1] 3 4 5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $group_names_t1</span></span>
<span><span class="co">#&gt; [1] "Russia"                 "Coastal West Alaska"    "Lower Yukon"           </span></span>
<span><span class="co">#&gt; [4] "Middle Yukon"           "Upper Yukon"            "North Alaska Peninsula"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $group_names_t2</span></span>
<span><span class="co">#&gt; [1] "Lower Yukon"  "Middle Yukon" "Upper Yukon" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $wildpops</span></span>
<span><span class="co">#&gt; [1] "CHBIG92.KIBIG93.KBIGB04.KBIGB95" "CHCRY92.KICRY94.KCRYA05"        </span></span>
<span><span class="co">#&gt; [3] "CHDMT92.KDEER94"                 "CHKAN92.KIKAN93.KKANE05"        </span></span>
<span><span class="co">#&gt; [5] "CHKOG92.KIKOG93.KKOGR05"         "CHNUU92.KINUS93"                </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $hatcheries</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="genetic-stock-identification">Genetic stock identification<a class="anchor" aria-label="anchor" href="#genetic-stock-identification"></a>
</h3>
<p>Once your input data set is ready, you can use
<code><a href="../reference/msgsi_mdl.html">msgsi_mdl()</a></code> to run the model. If you are used to running
<em>rubias</em>, <em>Ms.GSI</em> might feel a bit slow. That is because
1) we are running two GSI models in tandem, so it takes twice as long
than running a single model, and 2) <em>Ms.GSI</em> is written solely in
<em>R</em>, which is not as computationally efficient as language
<em>C</em>. So, why not code <em>Ms.GSI</em> in <em>C</em>? Because
we’re not technically advanced like the folks who developed
<em>rubias</em> package (i.e., we don’t know how to code in
<em>C++</em>).</p>
<p>Because of the running time, we recommend running the integrated
multistage model in conditional GSI mode (default setting). But there is
an option to run the model in fully Bayesian mode if one choose to. If
you run the model in fully Bayesian mode, you have the option to include
numbers of adaptation run. Some people think that adaptation run
encourages convergence in fully Bayesian mode. We have not test that
theory but provide the option for those who want to try it.</p>
<p>We demonstrate the model run with one chains of 150 iterations (first
50 as warm-up runs, or burn-ins). We only run one chain in this example
so it can pass CMD check while building the vignette document<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;The package we use to build &lt;em&gt;Ms.GSI&lt;/em&gt; has an
automated checking process. For some reason, automated check would fail
if we ran multiple chains in our examples. Probably something to do with
the parallel process we included in our function.&lt;/p&gt;"><sup>2</sup></a>. In
reality, you should <strong>definitely</strong> run multiple chains with
more iterations. There are also options to keep the burn-ins and set
random seed for reproducible results. We don’t show them in this example
though (but you can always <code><a href="../reference/msgsi_mdl.html">?msgsi_mdl</a></code>).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">msgsi_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/msgsi_mdl.html">msgsi_mdl</a></span><span class="op">(</span><span class="va">msgsi_dat</span>, nreps <span class="op">=</span> <span class="fl">150</span>, nburn <span class="op">=</span> <span class="fl">50</span>, thin <span class="op">=</span> <span class="fl">1</span>, nchains <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Running model (and the category is... Perfect Tens!)</span></span>
<span><span class="co">#&gt; Time difference of 2.706586 secs</span></span>
<span><span class="co">#&gt; June-08-2023 23:57</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summarizing-results">Summarizing results<a class="anchor" aria-label="anchor" href="#summarizing-results"></a>
</h3>
<div class="section level4">
<h4 id="stock-proportions">Stock proportions<a class="anchor" aria-label="anchor" href="#stock-proportions"></a>
</h4>
<p>The output of model contains eight items: <code>summ_t1</code>,
<code>trace_t1</code>, <code>summ_t2</code>, <code>trace_t2</code>,
<code>summ_comb</code>, <code>trace_comb</code>, <code>iden_t1</code>
and <code>idens_t2</code>. Items with “summ” are summary for reporting
group proportions and associated convergence diagnostics. If you want to
see summaries for stage one and two individually, <code>summ_t1</code>
and <code>summ_t2</code> will show you that. Most people probably want
to see the combined summary, <code>summ_comb</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">msgsi_out</span><span class="op">$</span><span class="va">summ_comb</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 8</span></span></span>
<span><span class="co">#&gt;    group                      mean  median      sd    ci.05   ci.95 GR     n_eff</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Russia                  2.27<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 2.15<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 1.59<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 6.92<span style="color: #949494;">e</span><span style="color: #BB0000;">- 6</span> 0.046<span style="text-decoration: underline;">5</span>  <span style="color: #BB0000;">NA</span>     27.0 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Coastal West Alaska     8.83<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 6.29<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 9.33<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 3.53<span style="color: #949494;">e</span><span style="color: #BB0000;">-11</span> 0.271   <span style="color: #BB0000;">NA</span>      1.86</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> North Alaska Peninsula  4.93<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 2.54<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> 1.06<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 7.18<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> 0.026<span style="text-decoration: underline;">2</span>  <span style="color: #BB0000;">NA</span>      6.46</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Northwest Gulf of Alas… 2.84<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 2.77<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 5.47<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 2.03<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 0.368   <span style="color: #BB0000;">NA</span>     21.5 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Copper                  7.26<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 2.14<span style="color: #949494;">e</span><span style="color: #BB0000;">-6</span> 1.95<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 7.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span> 0.004<span style="text-decoration: underline;">27</span> <span style="color: #BB0000;">NA</span>     31.9 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Northeast Gulf of Alas… 4.87<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 8.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-7</span> 1.52<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 2.55<span style="color: #949494;">e</span><span style="color: #BB0000;">-18</span> 0.002<span style="text-decoration: underline;">20</span> <span style="color: #BB0000;">NA</span>     55.7 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Coastal Southeast Alas… 9.32<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 2.06<span style="color: #949494;">e</span><span style="color: #BB0000;">-6</span> 3.25<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 2.90<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> 0.005<span style="text-decoration: underline;">20</span> <span style="color: #BB0000;">NA</span>     50.5 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> British Columbia        1.85<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 8.14<span style="color: #949494;">e</span><span style="color: #BB0000;">-8</span> 6.56<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 4.48<span style="color: #949494;">e</span><span style="color: #BB0000;">-18</span> 0.001<span style="text-decoration: underline;">15</span> <span style="color: #BB0000;">NA</span>    100   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> WA/OR/CA                4.87<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 2.21<span style="color: #949494;">e</span><span style="color: #BB0000;">-6</span> 1.26<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 2.85<span style="color: #949494;">e</span><span style="color: #BB0000;">-20</span> 0.002<span style="text-decoration: underline;">64</span> <span style="color: #BB0000;">NA</span>    100   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Lower Yukon             3.41<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 3.79<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 1.12<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 1.41<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 0.479   <span style="color: #BB0000;">NA</span>      1.16</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> Middle Yukon            7.63<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 7.44<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 2.19<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 4.41<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 0.113   <span style="color: #BB0000;">NA</span>     70.0 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> Upper Yukon             1.80<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 1.74<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 3.09<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 1.37<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 0.240   <span style="color: #BB0000;">NA</span>    100</span></span></code></pre></div>
<p>Most column names are self explanatory, but others might need some
additional descriptions. <code>ci.05</code> and <code>ci.95</code> are
the lower and upper bounds of 90% credible interval. <code>GR</code> is
the Gelman-Rubin statistic (a.k.a. <span class="math inline">\(\hat
R\)</span>). In this example, Gelman-Rubin statistic is not calculated
because we only run one chain. <code>n_eff</code> is the effective size,
or <span class="math inline">\(N_{eff}\)</span>. We will not discuss how
to diagnose convergence in this document. Please consult Gelman et
al. 2014, Gelman &amp; Rubin 1992, Brooks &amp; Gelman 1998 and other
literature on statistical methods.</p>
<p>Items with “trace” are the posterior sample history, or trace
history, for either stage one, two, or combined. Trace history is needed
for making trace plots. And if you need to combine reporting groups
proportions or combine variance, trace histories are what you need.
<code>trace_</code> items are tibbles with each reporting group as a
column. There are two additional columns, <code>itr</code> and
<code>chain</code>, to identify Markov chain Monte Carlo (MCMC) sampling
iteration and chain.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">msgsi_out</span><span class="op">$</span><span class="va">trace_comb</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 14</span></span></span>
<span><span class="co">#&gt;      Russia `Coastal West Alaska` North Alaska Peninsul…¹ Northwest Gulf of Al…²</span></span>
<span><span class="co">#&gt;       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 5.65<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span>                 0.175                7.99<span style="color: #949494;">e</span><span style="color: #BB0000;">- 8</span>                  0.346</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 2.67<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span>                 0.149                7.20<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span>                  0.288</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 3.98<span style="color: #949494;">e</span><span style="color: #BB0000;">-15</span>                 0.143                6.81<span style="color: #949494;">e</span><span style="color: #BB0000;">-10</span>                  0.260</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 9.83<span style="color: #949494;">e</span><span style="color: #BB0000;">- 7</span>                 0.107                3.19<span style="color: #949494;">e</span><span style="color: #BB0000;">-15</span>                  0.324</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 3.17<span style="color: #949494;">e</span><span style="color: #BB0000;">- 7</span>                 0.153                7.02<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span>                  0.336</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 1.64<span style="color: #949494;">e</span><span style="color: #BB0000;">-12</span>                 0.140                1.36<span style="color: #949494;">e</span><span style="color: #BB0000;">-14</span>                  0.339</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 7.23<span style="color: #949494;">e</span><span style="color: #BB0000;">- 6</span>                 0.137                1.74<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span>                  0.407</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 1.58<span style="color: #949494;">e</span><span style="color: #BB0000;">- 5</span>                 0.174                1.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span>                  0.450</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 8.47<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span>                 0.224                4.03<span style="color: #949494;">e</span><span style="color: #BB0000;">-15</span>                  0.352</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 3.00<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span>                 0.201                9.55<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span>                  0.392</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 90 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ abbreviated names: ¹​`North Alaska Peninsula`, ²​`Northwest Gulf of Alaska`</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 10 more variables: Copper &lt;dbl&gt;, `Northeast Gulf of Alaska` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Coastal Southeast Alaska` &lt;dbl&gt;, `British Columbia` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `WA/OR/CA` &lt;dbl&gt;, `Lower Yukon` &lt;dbl&gt;, `Middle Yukon` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Upper Yukon` &lt;dbl&gt;, itr &lt;dbl&gt;, chain &lt;dbl&gt;</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="individual-assignments">Individual assignments<a class="anchor" aria-label="anchor" href="#individual-assignments"></a>
</h4>
<p>The last two items in the output are the identity assignment history
of each individual in the mixture sample. Each column represents an
individual in the mixture, and each row records the identity assigned
during each iteration in each chain. These numbers are the population
identifiers in the same order as your population information files (and
baseline files). Individuals are ordered in the same as the input data
(i.e., mixture data).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">msgsi_out</span><span class="op">$</span><span class="va">idens_t1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 150</span></span></span>
<span><span class="co">#&gt;       V1    V2    V3    V4    V5    V6    V7    V8    V9   V10   V11   V12   V13</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>    11   192    11    56    97   152    97    95    32    56    84   192    97</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>    95    84    97    11    56   152    97   192    95    11    84    11    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>    84    87   192    95    84    11    97    32    11    11    97    84    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>    15    11    11   192    97   152    97   192    97    11    97    11    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>    11    11    11    32    39   152    97    87    11    11    97    97    56</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>    11   152    97    11    84    97    97    39    11    11    97    32    56</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>    11    87    97    87    56    11    97    11    56   192    11    84    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>    11    11    56    56    32   152    97    32    56    56    32    11    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>    11    56    32    32    56    84    97    39    87    87    84    84    84</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>    32    11   192    84    56   152   152    87    32    56    11    39    84</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 90 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 137 more variables: V14 &lt;int&gt;, V15 &lt;int&gt;, V16 &lt;int&gt;, V17 &lt;int&gt;, V18 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   V19 &lt;int&gt;, V20 &lt;int&gt;, V21 &lt;int&gt;, V22 &lt;int&gt;, V23 &lt;int&gt;, V24 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   V25 &lt;int&gt;, V26 &lt;int&gt;, V27 &lt;int&gt;, V28 &lt;int&gt;, V29 &lt;int&gt;, V30 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   V31 &lt;int&gt;, V32 &lt;int&gt;, V33 &lt;int&gt;, V34 &lt;int&gt;, V35 &lt;int&gt;, V36 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   V37 &lt;int&gt;, V38 &lt;int&gt;, V39 &lt;int&gt;, V40 &lt;int&gt;, V41 &lt;int&gt;, V42 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   V43 &lt;int&gt;, V44 &lt;int&gt;, V45 &lt;int&gt;, V46 &lt;int&gt;, V47 &lt;int&gt;, V48 &lt;int&gt;, …</span></span></span></code></pre></div>
<p>Individual identity output in this format may not be very useful for
most users. So, <code>Ms.GSI</code> has a function
<code><a href="../reference/indiv_assign.html">indiv_assign()</a></code> that summarize the reporting group
assignment probabilities for each individual in the mixture.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/indiv_assign.html">indiv_assign</a></span><span class="op">(</span>mdl_out <span class="op">=</span> <span class="va">msgsi_out</span>, mdl_dat <span class="op">=</span> <span class="va">msgsi_dat</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 150 × 5</span></span></span>
<span><span class="co">#&gt;    ID       `Not regional` `Lower Yukon` `Middle Yukon` `Upper Yukon`</span></span>
<span><span class="co">#&gt;  <span style="color: #BCBCBC;">*</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> fish_1             0.47          0.53              0             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> fish_10            0.42          0.58              0             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> fish_100           0.87          0.13              0             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> fish_101           0.5           0.5               0             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> fish_102           0.86          0.14              0             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> fish_103           0.11          0.89              0             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> fish_104           0.51          0.49              0             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> fish_105           0.78          0.22              0             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> fish_106           0.62          0.38              0             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> fish_107           0.38          0.62              0             0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 140 more rows</span></span></span></code></pre></div>
<p>The summary for individual assignment has a column named
<code>ID</code> that identifies each individual in the mixture.
<code>Not regional</code> column shows the probability of an individual
not belong to the group of interest, or, not from within region of
focus. The rest of the columns are probabilities for the regional
reporting groups. Probabilities in each row should sum up to one.</p>
</div>
<div class="section level4">
<h4 id="trace-plot">Trace plot<a class="anchor" aria-label="anchor" href="#trace-plot"></a>
</h4>
<p><em>Ms.GSI</em> has a function for you to make trace plot and examine
the mixing of MCMC chains.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/tr_plot.html">tr_plot</a></span><span class="op">(</span>obj <span class="op">=</span> <span class="va">msgsi_out</span><span class="op">$</span><span class="va">trace_comb</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="msgsi_vignette_files/figure-html/unnamed-chunk-14-1.png" alt="Trace plots for reporting group proportions." width="100%"><p class="caption">
Trace plots for reporting group proportions.
</p>
</div>
<p><span class="math display">\[\newcommand{\mx}[1]{\mathbf{#1}}\]</span></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="methods-math">Methods (math!)<a class="anchor" aria-label="anchor" href="#methods-math"></a>
</h2>
<div class="section level3">
<h3 id="pella-masuda-model">Pella-Masuda Model<a class="anchor" aria-label="anchor" href="#pella-masuda-model"></a>
</h3>
<p>This integrated multistage GSI model is essentially two Bayesian GSI
models stacked on top of each other; hence the name “multistage.” The
Pella-Masuda model (Pella &amp; Masuda 2001) is the Bayesian GSI model
that make up each stage in the integrated multistage model. We will
first describe the Pella-Masuda model before discussing the development
of a integrated multistage model.</p>
<p>In a group of mixed populations, Pella-Masuda model assigns
population identities to each individual based on its genetic make-up
(e.g. genotype). Then the model estimates the overall population
proportions based on the numbers of individuals assigned to each
population. In the fishery context, genetic data of the individuals is
called the mixture sample because it consists multi-locus genotype of
individual fish collected from a mixed-stock fishery. <span class="math inline">\(\mx x\)</span> denotes the mixture sample. In this
document, a bold-font letter represents a <em>number set</em>, or a
collection of distinct elements. For example, <span class="math inline">\(\mx x\)</span> is a set that contains individual
<span class="math inline">\(x\)</span> elements. And <span class="math inline">\(x_{m,l,j}\)</span> is the count of allele <span class="math inline">\(j\)</span> in locus <span class="math inline">\(l\)</span> for individual fish <span class="math inline">\(m\)</span>, where <span class="math inline">\(m
\in \{1,2,...,M\}\)</span>, <span class="math inline">\(l \in
\{1,2,...,L\}\)</span>, and <span class="math inline">\(j \in
\{1,2,...,J_l\}\)</span> depends on locus <span class="math inline">\(l\)</span>.</p>
<p>Genetic data of the populations is called the baseline sample because
it consists genotype compositions of various baseline populations
collected at their spawning locations. Researchers select sampling
locations to best represent the populations in an area. <span class="math inline">\(\mx y\)</span> denotes the baseline sample. <span class="math inline">\(y_{k,l,j}\)</span> is the count of allele <span class="math inline">\(j\)</span> in locus <span class="math inline">\(l\)</span> for a sample of size <span class="math inline">\(n_{k,l}\)</span> collected from baseline
population <span class="math inline">\(k\)</span>, where <span class="math inline">\(k \in \{1,2,...,K\}\)</span>.</p>
<p>For both mixture and baseline samples, it is assumed that allele
counts in each locus follow a multinomial distribution<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;What is a multinomial distribution? They are the counts
of multiple categories based on their corresponding probabilities. For
example, we randomly toss 10 Swedish Fish to three kids. Each time we
toss a fish, the probability of kid #1 catching it is 0.3, kid #2, 0.3,
and kid #3, 0.4. The numbers of fish caught by the three kids are
multinomially distributed with probabilities = {0.3, 0.3, 0.4} and a
total size 10.&lt;/p&gt;"><sup>3</sup></a>. Using another made-up
example, in a baseline sample, there are two allele types in locus 1 for
population 2. Counts for the two alleles are <span class="math inline">\(y_{1,2,1}, y_{1,2,2}\)</span>, and they follow a
multinomial distribution with parameters <span class="math inline">\(q_{1,2,1}, q_{1,2,2}\)</span> and size <span class="math inline">\(n_{2,1}\)</span>. Note that <span class="math inline">\(q_{1,2,1}, q_{1,2,2}\)</span> are the relative
frequencies of the two alleles in locus 1 for population 2. In a
Bayesian framework, we need to specify prior distributions for
parameters; therefore, we place a Dirichlet<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;What is a Dirichlet distribution? Using the same example
for multinomial distribution, the fish catching probabilities of all
three kids are Dirichlet-distributed with fish counts of all three kids
as parameters. It is worth noting that the parameters do not have to be
integers as long as they are &amp;gt; 0.&lt;/p&gt;"><sup>4</sup></a> prior distribution on
<span class="math inline">\(q_{1,2,1}, q_{1,2,2}\)</span> with
hyperparameters<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;A hyperparameter is a parameter of a prior
distribution.&lt;/p&gt;"><sup>5</sup></a> <span class="math inline">\(\beta_{1,1},
\beta_{1,2}\)</span>. Usually we set the priors to be equal for all
loci. In this example, let <span class="math inline">\(\beta_{1,1} =
\beta_{1,2} = 1/2\)</span> based on the number of alleles for locus
1.</p>
<p><span class="math inline">\(\mx q\)</span> represents <span class="math inline">\(q_{1,2,1}\)</span> and <span class="math inline">\(q_{1,2,2}\)</span>, together with allele
frequencies of other loci and other populations. As you can see, <span class="math inline">\(\mx q\)</span> and <span class="math inline">\(\mx
y\)</span> have the same dimension because each relative frequency
corresponds to an allele count. In the model, allele frequencies of
baseline populations, <span class="math inline">\(\mx q\)</span>,
determine population proportions. And population proportions determines
the identities of individual fish. Individual identities are then
tallied and summarized to update baseline allele frequencies. <span class="math inline">\(\mx y\)</span> can be expressed as follows:</p>
<p><span class="math inline">\(\mx y_k \sim Mult(\mx n_k, \mx
q_k)\)</span></p>
<p>Prior distribution for <span class="math inline">\(\mx
q\)</span>:</p>
<p><span class="math inline">\(\mx q_k \sim Dirich(\mx
\beta)\)</span>,</p>
<p>where <span class="math inline">\(\mx \beta = 1/J_l\)</span></p>
<p>For mixture sample, allele counts in each locus of individual fish
also follows a multinomial distributions. If a fish came from a certain
population, its distribution of allele counts should resemble the allele
frequencies of the baseline population which it came from. However, the
identity of the individual fish is unknown so it needs to be estimated.
Here we let <span class="math inline">\(\mx z_m\)</span> represent the
population identify for the <span class="math inline">\(m\)</span><sup>th</sup> mixture individual. <span class="math inline">\(\mx z_m\)</span> is composed of 0’s and an 1 with
a length <span class="math inline">\(K\)</span> (e.g. number of baseline
populations). <span class="math inline">\(z_{m,k} = 1\)</span> if
individual <span class="math inline">\(m\)</span> belongs to population
<span class="math inline">\(k\)</span>, and <span class="math inline">\(z_{m,k} = 0\)</span> otherwise. In a made-up
example, <span class="math inline">\(\mx z_{100} = \{0, 0, 1, 0,
0\}\)</span> means that there are five baseline populations, and
individual fish #100 comes from population 3.</p>
<p>We place a multinomial prior on <span class="math inline">\(z_{m,1},
z_{m,2}, ..., z_{m,K}\)</span> with size 1 and probabilities equal to
population proportions <span class="math inline">\(p_1, p_2, ...,
p_K\)</span>. We specify a Dirichlet prior distribution on <span class="math inline">\(p_1, p_2, ..., p_K\)</span> with hyperparameters
<span class="math inline">\(\alpha_1, \alpha_2, ..., \alpha_K\)</span>,
where <span class="math inline">\(\alpha_1 = \alpha_2 = ... = \alpha_K =
1/K\)</span>. We usually set <span class="math inline">\(\alpha\)</span>
to be equal for all reporting groups, but they can be set based on prior
knowledge in population proportions. We express <span class="math inline">\(\mx z\)</span> as follows:</p>
<p><span class="math inline">\(\mx z_m \sim Mult(\mx 1, \mx
p)\)</span></p>
<p>Prior distribution for <span class="math inline">\(\mx
p\)</span>:</p>
<p><span class="math inline">\(\mx p \sim Dirich(\mx
\alpha)\)</span>,</p>
<p>where <span class="math inline">\(\mx \alpha = 1/K\)</span></p>
<p>As mentioned, for mixture sample, allele counts in each locus of
individual fish follows a multinomial distributions. The parameters are
allele frequencies of the corresponding baseline population with size
the numbers of ploidy for each respective locus. Remember that
population identity <span class="math inline">\(z_{m,k} = 1\)</span> if
individual <span class="math inline">\(m\)</span> belongs to population
<span class="math inline">\(k\)</span>, and <span class="math inline">\(z_{m,k} = 0\)</span> otherwise. When multiplying
population identities, <span class="math inline">\(z_{m,1}, z_{m,2},
..., z_{m,K}\)</span>, and allele frequencies of baseline populations,
<span class="math inline">\(\mx q_1, \mx q_2, ..., \mx q_K\)</span>,
only allele frequencies of baseline population which individual <span class="math inline">\(m\)</span> belong to would remain while the rest
goes to zero. <span class="math inline">\(\mx x\)</span> is expressed
below. <span class="math inline">\(\mx{ploidy} = ploidy_1, ploidy_2,
..., ploidy_L\)</span> denotes ploidy for each locus.</p>
<p><span class="math inline">\(\mx x_m \sim Mult(\mx{ploidy}, \mx z_m
\cdot \mx q)\)</span></p>
<p>Moran and Anderson (2018) implement a genetic mixture analysis as a
<em>R</em> package, <em>rubias</em>. Their program has been widely used
by researchers around the world, including here at the GCL.
<em>rubias</em> utilizes a model structure called the conditional
genetic stock identification model, or the conditional GSI model, that
is modified from the Pella-Masuda model. The main difference between the
two models is that, in the conditional model, <span class="math inline">\(\mx q\)</span> is integrated out of the
distribution of mixture sample, <span class="math inline">\(\mx
x_m\)</span>. That is, baseline allele frequencies are not being updated
in the model. The result of that, <span class="math inline">\(\mx
x_m\)</span> takes a form of a compound Dirichlet-multinomial
distribution (Johnson at el. 1997):</p>
<p><span class="math inline">\(\mx x_m \sim CDM(\mx{ploidy}, \mx z_m
\cdot \mx v)\)</span>,</p>
<p>where <span class="math inline">\(\mx v\)</span> is <span class="math inline">\(\mx \beta + \mx y\)</span>. We are not going to
attempt proving the theory behind the conditional model in this document
(details can be found in Moran &amp; Anderson 2018). But since <span class="math inline">\(\mx q\)</span> has been integrated out of <span class="math inline">\(\mx x_m\)</span>, the process for estimating
parameters is simpler and more streamlined. We have implemented
conditional GSI in each stage of our integrated multistage model.</p>
</div>
<div class="section level3">
<h3 id="extend-to-multistage">Extend to multistage<a class="anchor" aria-label="anchor" href="#extend-to-multistage"></a>
</h3>
<p>In the integrated multistage framework, the two Pella-Masuda models
are connected because a fish being assigned to a regional group at the
second stage is conditional based on whether that fish is assigned to a
reporting group in the regional baseline during the first stage. The
following describes the conditional relationship between the broad-scale
and the regional baselines:</p>
<p><span class="math inline">\(\mx z^{(regional)}_m | z^{(broad)}_m =
\begin{cases} \mx z^{(regional)}_m &amp; \text{if } z^{(broad)}_m = 1\\
\mx 0 &amp; \text{otherwise} \end{cases}\)</span></p>
<p><span class="math inline">\(\mx z^{(regional)}_m\)</span> are the
same as described in the previous section for a single stage scenario.
They are vectors of indicators (<span class="math inline">\(0\)</span>
or <span class="math inline">\(1\)</span>) identifying the regional
population that individual <span class="math inline">\(m\)</span>
belongs to. <span class="math inline">\(z^{(broad)}_m\)</span> is an
indicator as well, and <span class="math inline">\(z^{(broad)}_m =
1\)</span> if individual <span class="math inline">\(m\)</span> belongs
to a reporting group within the regional baseline and <span class="math inline">\(0\)</span> otherwise. Combining <span class="math inline">\(z^{(regional)}\)</span> and <span class="math inline">\(z^{(broad)}\)</span>, <span class="math inline">\(z^{(regional)}_{m,k} | z^{(broad)}_m = 1\)</span>
if individual <span class="math inline">\(m\)</span> belongs to a
reporting group within the regional baseline and is assigned to
population <span class="math inline">\(k\)</span> in the regional
baseline.</p>
<p>Individually, vectors <span class="math inline">\(Z^{(regional)}_m\)</span> have a multinomial
distribution, <span class="math inline">\(Z^{(regional)}_m \sim Mult(1,
\pi_m)\)</span>, where <span class="math inline">\(\pi_m\)</span> are
vectors of probabilities for individual <span class="math inline">\(m\)</span> being assigned to each regional
reporting group. The prior distribution for <span class="math inline">\(\pi_m\)</span> is a Dirichlet, <span class="math inline">\(Dirich(K^{-1})\)</span>, where <span class="math inline">\(K\)</span> is the number of reporting groups in
the regional baseline. The posterior distribution for <span class="math inline">\(\pi_m|z^{(regional)}_m\)</span> is also a
Dirichlet, <span class="math inline">\(Dirich(\sum_{m=1}^M
z^{(regional)}_{m,1} + K^{-1}, \sum_{m=1}^M z^{(regional)}_{m,2} +
K^{-1},..., \sum_{m=1}^M z^{(regional)}_{m,K} + K^{-1})\)</span>. We
will describe in more details on deriving the posteriors in the next
section.</p>
<p>In the context of the integrated multistage framework, we only care
that, during the first stage, whether an individual is assigned to a
reporting group within the regional baseline; therefore, individual
identity has a binomial outcome, and <span class="math inline">\(Z^{(broad)}_m \sim Bin(1, \rho_m)\)</span>. The
prior for <span class="math inline">\(\rho_m\)</span> has a Beta
distribution with two scale parameters; one equals to the proportions of
broad-scale stocks in the regional baseline, and the other equals to the
proportions of broad-scale stocks outside of the regional baseline. The
expectation for <span class="math inline">\(\rho_m\)</span> equals to
the proportion of broad-scale stocks within the regional baseline. The
posterior distribution for <span class="math inline">\(\rho_m|z^{(broad)}_m\)</span> is also a Beta, with
the first parameter equals to the sum of individuals assigned to a
broad-scale stock in the regional baseline and the proportion of
broad-scale stocks in the regional baseline, and the second parameter
equals to the sum of individuals assigned to a broad-scale stock outside
of the regional baseline and the proportion of broad-scale stocks
outside of the regional baseline.</p>
<p>Probability of an individual being assigned to regional population
<span class="math inline">\(k\)</span> can be derived using the Bayes
rule:</p>
<p><span class="math display">\[\begin{aligned}
P&amp;(Z^{(regional)}_{m,k} | Z^{(broad)}_m) =\\
&amp;\frac{P(Z^{(broad)}_m | Z^{(regional)}_{m,k}) \cdot
P(Z^{(regional)}_{m,k})} {\sum_{k=1}^{K} P(Z^{(broad)}_m |
Z^{(regional)}_{m,k}) \cdot P(Z^{(regional)}_{m,k})} =
\frac{P(Z^{(broad)}_m) \cdot P(Z^{(regional)}_{m,k})}{P(Z^{(broad)}_m)}
\end{aligned}\]</span></p>
<p>This can be simplified into <span class="math inline">\(P(Z^{(regional)}_{m,k})\)</span>.</p>
<!-- Summing up the individual assignment probabilities we get the weighted average of assignment probability to regional population $k$ for all individuals: -->
<!-- $$\begin{aligned} -->
<!-- P&(Z^{(regional)}_k | Z^{(broad)}) =\\ &\sum\limits_{m=1}^{M} \frac{P(Z^{(broad)}_m | Z^{(regional)}_{m,k}) \cdot P(Z^{(regional)}_{m,k})} {\sum_{k=1}^{K} P(Z^{(broad)}_m | Z^{(regional)}_{m,k}) \cdot P(Z^{(regional)}_{m,k})} = \frac{\sum_{m=1}^{M}P(Z^{(broad)}_m) \cdot P(Z^{(regional)}_{m,k})}{\sum_{m=1}^{M} P(Z^{(broad)}_m)} -->
<!-- \end{aligned}$$ -->
<p>During a GSI analysis, two Pella-Masuda models proceed in tandem,
each with its own baseline maker suite but the same individuals in the
mixture sample.</p>
</div>
<div class="section level3">
<h3 id="gibbs-sampler-where-the-fun-go-round-and-round">Gibbs Sampler: where the fun go round and round<a class="anchor" aria-label="anchor" href="#gibbs-sampler-where-the-fun-go-round-and-round"></a>
</h3>
<p>Deriving the values of parameters in each stage of the integrated
multistage model requires finding the joint posterior distribution of
Pella-Masuda model in each stage, <span class="math inline">\(\mx p, \mx
q,\mx z, \mx y, \mx\alpha, \mx\beta\)</span>. In this section, we will
introduce the concepts and algorithm to sample from this posterior
distribution in a single Pella-Masuda model, which then can be extend to
a integrated multistage framework.</p>
<p>Gibbs sampler is a type of MCMC methods that sequentially sample
parameter values from a Markov chain. With enough sampling, the Markov
chain will eventually converge to the desire distribution of interest.
The most appealing quality of Gibbs sampler is its reduction of a
multivariate problem (such as Pella-Masuda model) to a series of more
manageable lower-dimensional problems. A full description of Gibbs
sampler and MCMC methods is beyond the scope of this document; however,
further information can be found in numerous resources devoting to
Bayesian data analysis (see Carlin &amp; Louis 2009; Robert &amp;
Casella 2010; Gelman et al. 2014)</p>
<p>To illustrate, suppose we would like to determine the joint posterior
distribution of interest, <span class="math inline">\(p(\mx \theta|\mx
y)\)</span>, where <span class="math inline">\(\mx \theta = \{\theta_1,
\theta_2,..., \theta_K\}\)</span>. Most likely the multivariate <span class="math inline">\(p(\mx \theta|\mx y)\)</span> would be too
complicated to sample from. However, if we can figure out how to break
up the joint posterior distribution into individual full conditional
distributions, each parameter in <span class="math inline">\(\mx
\theta\)</span> can be sampled one by one sequentially using a Gibbs
sampler algorithm. The process begins with an arbitrary set of starting
values <span class="math inline">\(\theta^{(0)}_2, \theta^{(0)}_3,...,
\theta^{(0)}_K\)</span> and proceeds as follows:</p>
<p>For <span class="math inline">\(t = 1,2,...,T\)</span>, repeat</p>
<ol style="list-style-type: decimal">
<li><p>Draw <span class="math inline">\(\theta^{(t)}_1\)</span> from
<span class="math inline">\(p(\theta_1|\theta^{(t-1)}_2,
\theta^{(t-1)}_3,..., \theta^{(t-1)}_k, \mx y)\)</span></p></li>
<li>
<p>Draw <span class="math inline">\(\theta^{(t)}_2\)</span> from
<span class="math inline">\(p(\theta_2|\theta^{(t)}_1,
\theta^{(t-1)}_3,..., \theta^{(t-1)}_k, \mx y)\)</span></p>
<p>⋮</p>
</li>
</ol>
<!-- --><ol start="11" style="list-style-type: lower-alpha">
<li>Draw <span class="math inline">\(\theta^{(t)}_k\)</span> from <span class="math inline">\(p(\theta_k|\theta^{(t)}_1, \theta^{(t)}_2,...,
\theta^{(t)}_{k-1}, \mx y)\)</span>
</li>
</ol>
<p>This would work best if the full conditionals are some known
distributions that we can easily sample from (although it’s not
required). In our case with Pella-Masuda model, we rely on two main
concepts, the Bayes theorem and conjugacy, to do the trick. Briefly, for
estimating parameters <span class="math inline">\(\mx\theta\)</span>
from data <span class="math inline">\(\mx D\)</span>, according to Bayes
Rule, <span class="math inline">\(p(\mx\theta|\mx D) = \displaystyle
\frac{p(\mx D|\mx\theta)p(\mx\theta)}{p(\mx D)}\)</span>. <span class="math inline">\(p(\mx\theta|\mx D)\)</span> is the joint posterior
distribution for parameters <span class="math inline">\(\mx\theta\)</span>, <span class="math inline">\(p(\mx D|\mx\theta)\)</span> is the likelihood of
observing the data given the parameters, <span class="math inline">\(p(\mx\theta)\)</span> is the prior distribution of
the parameters, and <span class="math inline">\(p(\mx D)\)</span> is the
constant marginal distribution of the data. <span class="math inline">\(p(\mx D)\)</span> is often mathematically
difficult to obtain; however, because <span class="math inline">\(p(\mx
D)\)</span> is a constant number, we can ignore it by reducing the
posterior distribution to <span class="math inline">\(p(\mx\theta|\mx D)
\propto p(\mx D|\mx\theta)p(\mx\theta)\)</span>.</p>
<p>So, how does Bayes Rule help us estimating parameters in the
Pella-Masuda model? First, the joint posterior distribution has to be
split up into smaller pieces. That is, we separate the joint posterior
into likelihood of the data and priors for the parameters:</p>
<p><span class="math inline">\(p(\mx p, \mx q, \mx z, \mx y, \mx\alpha,
\mx\beta)\)</span></p>
<p><span class="math inline">\(\propto p(\mx x|\mx z, \mx q) p(\mx y|\mx
q) \cdot p(\mx p|\mx\alpha) p(\mx q|\mx\beta) p(\mx z|\mx
p)\)</span></p>
<p>With some re-arrangements and hand-waving, we arrive at the marginal
posterior distributions for <span class="math inline">\(\mx q\)</span>
and <span class="math inline">\(\mx p\)</span>$:</p>
<p><span class="math inline">\(p(\mx x|\mx z, \mx q) p(\mx y|\mx q)
\cdot p(\mx p|\mx\alpha) p(\mx q|\mx\beta) p(\mx z|\mx p)\)</span></p>
<p><span class="math inline">\(= p(\mx x|\mx z, \mx q) p(\mx y|\mx q)
p(\mx q|\mx\beta) \cdot p(\mx z|\mx p) p(\mx p|\mx\alpha)\)</span></p>
<p><span class="math inline">\(\propto p(\mx x,\mx y,\mx z|\mx q) p(\mx
q|\mx\beta) \cdot p(\mx z|\mx p) p(\mx p|\mx\alpha)\)</span></p>
<p><span class="math inline">\(\propto p(\mx q|\mx x,\mx y,\mx
z,\mx\beta) \cdot p(\mx p|\mx z,\mx\alpha)\)</span></p>
<p>Next, we take advantage of a mathematical property called
<em>conjugacy</em> to help us determine the marginal posterior
distributions. Based on this property, the posterior distribution
follows the same parametric form as the prior distribution when prior is
a <em>conjugate family</em> for the likelihood. For example, if the
likelihood of data is binomial distribution and the prior of parameter
is beta distribution, then the posterior is also beta distribution
because beta is a conjugate family for binomial. There are many
conjugate families, and Dirichlet and multinomial are another
example.</p>
<p>Utilizing conjugacy property, we will determine each of the marginal
posterior distributions for <span class="math inline">\(\mx q\)</span>
and <span class="math inline">\(\mx p\)</span>.</p>
<div class="section level4">
<h4 id="marginal-posterior-pqx-y-z-beta">Marginal Posterior p(q|x, y, z, <span class="math inline">\(\beta\)</span>)<a class="anchor" aria-label="anchor" href="#marginal-posterior-pqx-y-z-beta"></a>
</h4>
<p>We determine that posterior <span class="math inline">\(p(\mx q|\mx
x,\mx y,\mx z,\mx\beta)\)</span> is Dirichlet-distributed because
Dirichlet prior <span class="math inline">\(p(\mx q|\mx\beta)\)</span>
is a conjugate family for the multinomial likelihoods <span class="math inline">\(p(\mx x|\mx z, \mx q)\)</span> and <span class="math inline">\(p(\mx y|\mx q)\)</span>. To determine the exact
parameterization for the posterior distribution, we need to derive the
prior and likelihoods first.</p>
<p>Likelihood <span class="math inline">\(p(\mx x|\mx z, \mx q)\)</span>
can be derived in two steps. The first step we conditioned the
likelihood on <span class="math inline">\(\mx z\)</span> so that</p>
<p><span class="math inline">\(p(\mx x|\mx z, \mx q) \propto
\displaystyle \prod^{M}_{m=1} \prod^{K}_{k=1} [f(\mx x_m|\mx
q_k)]^{z_{m,k}}\)</span>,</p>
<p>where <span class="math inline">\(f(\mx x_m|\mx q_k)\)</span> is the
relative frequency of multi-locus genotype for individual <span class="math inline">\(m\)</span> in population <span class="math inline">\(k\)</span>. In the next step, we derive <span class="math inline">\(f(\mx x_m|\mx q_k)\)</span>:</p>
<p><span class="math inline">\(f(\mx x_m|\mx q_k) \propto \displaystyle
\prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{x_{m,l,j}}_{k,l,j}\)</span></p>
<p>Then we combine the two,</p>
<p><span class="math inline">\(p(\mx x|\mx z, \mx q) \propto
\displaystyle \prod^{M}_{m=1} \prod^{K}_{k=1} [f(\mx x_m|\mx
q_k)]^{z_{m,k}}\)</span></p>
<p><span class="math inline">\(\propto \displaystyle \prod^{M}_{m=1}
\prod^{K}_{k=1} [\displaystyle \prod^{L}_{l=1} \prod^{J_l}_{j=1}
q^{x_{m,l,j} \cdot z_{m,k}}_{k,l,j}]\)</span></p>
<p><span class="math inline">\(\propto \displaystyle \prod^{K}_{k=1}
\prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\sum^{M}_{m=1} (x_{m,l,j} \cdot
z_{m,k})}_{k,l,j}\)</span></p>
<p>Deriving likelihood <span class="math inline">\(p(\mx y|\mx
q)\)</span> is more straightforward. It is the product of relative
frequency of multi-locus genotype for each population:</p>
<p><span class="math inline">\(p(\mx y|\mx q) \propto \displaystyle
\prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1}
q^{y_{k,l,j}}_{k,l,j}\)</span></p>
<p>And <span class="math inline">\(p(q|\mx\beta)\)</span> is Dirichlet
prior distribution. Its probability density has a kernel<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;Without involving too much math, &lt;em&gt;kernel&lt;/em&gt; here
refers to what is left over after factoring out the constant from a
probability density function (PDF). For example, The PDF of Dirichlet
distribution is &lt;span class="math inline"&gt;\(\frac{1}{B(\mx \alpha)}
\displaystyle \prod^K_{i=1} x^{\alpha_i - 1}_i\)&lt;/span&gt;. Beta function
&lt;span class="math inline"&gt;\(B(\mx \alpha)\)&lt;/span&gt; can be factored out,
and &lt;span class="math inline"&gt;\(\displaystyle \prod^K_{i=1} x^{\alpha_i
- 1}_i\)&lt;/span&gt; is the kernel.&lt;/p&gt;'><sup>6</sup></a> of <span class="math inline">\(\mx q^{\mx \beta - 1}\)</span>. We can express the
likelihood as</p>
<p><span class="math inline">\(p(\mx q|\mx\beta) \propto \displaystyle
\prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\beta_{l,j} -
1}_{k,l,j}\)</span>.</p>
<p>Put all the likelihoods together,</p>
<p><span class="math inline">\(p(\mx q|\mx x,\mx y,\mx z,\mx\beta)
\propto p(\mx x|\mx z, \mx q) p(\mx y|\mx q) p(\mx
q|\mx\beta)\)</span></p>
<p><span class="math inline">\(\propto \displaystyle \prod^{K}_{k=1}
\prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\sum^{M}_{m=1} (x_{m,l,j} \cdot
z_{m,k})}_{k,l,j} \cdot \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1}
\prod^{J_l}_{j=1} q^{y_{k,l,j}}_{k,l,j} \cdot \displaystyle
\prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\beta_{l,j} -
1}_{k,l,j}\)</span></p>
<p><span class="math inline">\(= \displaystyle \prod^{K}_{k=1}
\prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\sum^{M}_{m=1} (x_{m,l,j} \cdot
z_{m,k}) + y_{k,l,j} + \beta_{l,j} - 1}_{k,l,j}\)</span></p>
<p>It is <em>elementary</em> for anybody to recognize that <span class="math inline">\(\displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1}
\prod^{J_l}_{j=1} q^{\sum^{M}_{m=1} (x_{m,l,j} \cdot z_{m,k}) +
y_{k,l,j} + \beta_{l,j} - 1}_{k,l,j}\)</span> is the kernel for
Dirichlet distribution. Hence,</p>
<p><span class="math inline">\(\mx q_{k,l}|\mx x,\mx y,\mx z,\mx\beta
\sim Dirich(\displaystyle \sum^{M}_{m=1} x_{m,l,j} z_{m,k} + y_{k,l,j} +
\beta_{l,j})\)</span></p>
</div>
<div class="section level4">
<h4 id="marginal-posterior-ppz-alpha">Marginal Posterior p(p|z, <span class="math inline">\(\alpha\)</span>)<a class="anchor" aria-label="anchor" href="#marginal-posterior-ppz-alpha"></a>
</h4>
<p>Using the same logic as previously, <span class="math inline">\(p(\mx
p|\mx z,\mx\alpha)\)</span> is also Dirichlet-distributed due to a
Dirichlet prior <span class="math inline">\(p(\mx p|\mx\alpha)\)</span>
and a multinomial likelihood <span class="math inline">\(p(\mx z|\mx
p)\)</span>.</p>
<p><span class="math inline">\(p(\mx p|\mx z,\mx\alpha) \propto p(\mx
z|\mx p) p(\mx p|\mx\alpha)\)</span></p>
<p><span class="math inline">\(\propto \displaystyle \prod^{M}_{m=1}
\prod^{K}_{k=1} p^{z_{m,k}}_k \cdot \prod^{K}_{k=1} p^{\alpha_k -
1}_k\)</span></p>
<p><span class="math inline">\(\propto \displaystyle \prod^{K}_{k=1}
p^{\sum^M_{m=1}z_{m,k} + \alpha_k - 1}_k\)</span></p>
<p>Once again, we recognize it as the kernel for Dirichlet
distribution:</p>
<p><span class="math inline">\(\mx p|\mx z,\mx\alpha \sim
Dirich(\displaystyle \sum^M_{m=1}z_{m,k} + \alpha_k)\)</span></p>
</div>
<div class="section level4">
<h4 id="algorithm">Algorithm<a class="anchor" aria-label="anchor" href="#algorithm"></a>
</h4>
<p>There is one more distribution to figure out before we can start our
Gibbs sampler routine (and you thought we’re all set, lol). We would
need to know how to sample <span class="math inline">\(\mx z_m|\mx p,
\mx q, \mx x_m\)</span>, the population identity for individual fish
<span class="math inline">\(m\)</span> (in components 1 and 2) given the
population proportions and genotype. If the probability of fish <span class="math inline">\(m\)</span> belong to population <span class="math inline">\(k\)</span> is <span class="math inline">\(p_k\)</span>, and the likelihood of observing
relative frequency of genotype for fish <span class="math inline">\(m\)</span> in population <span class="math inline">\(k\)</span> is <span class="math inline">\(f(\mx
x_m|\mx q_k)\)</span>, then the probability of fish <span class="math inline">\(m\)</span> belong to population <span class="math inline">\(k\)</span> given the population proportions and
genotype is <span class="math inline">\(\displaystyle \frac{p_k \cdot
f(\mx x_m|\mx q_k)}{\sum^K_{k'=1}p_{k'} \cdot f(\mx x_m|\mx
q_{k'})}\)</span>. The denominator should sum to one, so we only
need to calculate the numerator.</p>
<p><span class="math inline">\(\mx z_m|\mx p, \mx q, \mx x_m\)</span>
has the following distribution:</p>
<p><span class="math inline">\(\mx z_m|\mx p, \mx q, \mx x_m \sim
Mult(1, \mx{p'}_m)\)</span>,</p>
<p>where <span class="math inline">\(p'_{m,k} = p_k \cdot f(\mx
x_m|\mx q_k)\)</span>. We draw the initial values for <span class="math inline">\(\mx q_k\)</span> based on its prior
distribution.</p>
<p>Once we figured out all the pieces in the Gibbs sampler, we may begin
the process with starting values for <span class="math inline">\(\mx
p^{(0)}\)</span> and <span class="math inline">\(\mx q^{(0)}\)</span>.
We proceed as follows:</p>
<p>For <span class="math inline">\(t = 1,2,...,T\)</span>, repeat</p>
<ol style="list-style-type: decimal">
<li><p>Determine the population identity of mixture individuals, <span class="math inline">\(\mx z^{(t)}_m|\mx p^{(t-1)}, \mx q^{(t-1)}, \mx
x_m \sim Mult(1, \mx{p'}_m)\)</span>.</p></li>
<li><p>Draw marginal posterior distributions for <span class="math inline">\(\mx q^{(t)}\)</span> and <span class="math inline">\(\mx p^{(t)}\)</span> from <span class="math inline">\(p(\mx q|\mx x,\mx y,\mx z^{(t)},\mx\beta)\)</span>
and <span class="math inline">\(p(\mx p|\mx z^{(t)},\mx\alpha)\)</span>
respectively.</p></li>
</ol>
<p><span class="math inline">\(T\)</span> should be large enough to
ensure the sampler chain converges to desire distribution of interest.
Usually it takes thousands of iterations.</p>
<p>To extend the algorithm to multistage:</p>
<ol style="list-style-type: decimal">
<li><p>Determine the population identities of mixture individuals at the
first stage, <span class="math inline">\(\mx z^{(broad)(t)}_m|\mx
p^{(broad)(t-1)}, \mx q^{(broad)(t-1)}, \mx x_m \sim Mult(1,
\mx{p'}^{(broad)}_m)\)</span>.</p></li>
<li><p>Determine the population identities of mixture individuals at the
second stage, <span class="math inline">\(\mx
z^{(regional)(t)}_m|z^{(broad)(t)}_m, \mx p^{(regional)(t-1)}, \mx
q^{(regional)(t-1)}, \mx x_m \sim Mult(1,
\mx{p'}^{(regional)}_m)\)</span>.</p></li>
<li><p>Draw marginal posterior distributions for <span class="math inline">\(\mx q^{(broad)(t)}\)</span>, <span class="math inline">\(\mx p^{(broad)(t)}\)</span>, <span class="math inline">\(\mx q^{(regional)(t)}\)</span> and <span class="math inline">\(\mx p^{(regional)(t)}\)</span> from <span class="math inline">\(p(\mx q^{(broad)}|\mx x,\mx y^{(broad)},\mx
z^{(broad)(t)},\mx\beta^{(broad)})\)</span>, <span class="math inline">\(p(\mx p^{(broad)}|\mx
z^{(broad)(t)},\mx\alpha^{(broad)})\)</span>, <span class="math inline">\(p(\mx q^{(regional)}|\mx x,\mx y^{(regional)},\mx
z^{(regional)(t)},\mx\beta^{(regional)})\)</span> and <span class="math inline">\(p(\mx p^{(regional)}|\mx
z^{(regional)(t)},\mx\alpha^{(regional)})\)</span>
respectively.</p></li>
</ol>
<p>Steps 1 through 3 are repeated until simulations converge to a desire
distribution.</p>
<p>Implementing the conditional GSI model only requires a slight
modification from the above algorithm. At the first stage, <span class="math inline">\(f(\mx x_m|\mx q^{(broad)}_k)\)</span> will only
need to be derived once in the beginning of the process, and <span class="math inline">\(\mx q^{(broad)}\)</span> is no longer updated in
step 3. At the second stage, <span class="math inline">\(\mx
x_m\)</span> changes depending on the outcome of <span class="math inline">\(z^{(broad)}_m\)</span>; therefore, <span class="math inline">\(f(\mx x_m|\mx q^{(regional)}_k)\)</span> is
recalculated. However, <span class="math inline">\(\mx
q^{(regional)}\)</span> does not need to be updated. Everything else in
the algorithm stays the same.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Brooks, S. P., and A. Gelman. 1998. General methods for monitoring
convergence of iterative simulations. <em>Journal of Computational and
Graphical Statistics</em>. 7:434–455.</p>
<p>Carlin, B. and T. Louis. 2009. <em>Bayesian Methods for Data
Analysis, 3rd Edition</em>. CRC Press. New York.</p>
<p>Gelman, A., and D. B. Rubin. 1992. Inference from iterative
simulation using multiple sequences. <em>Statistical Science</em>.
7:457–472.</p>
<p>Gelman, A., J. Carlin, H. Stern, D. Dunson, A. Vehtari and D. Rubin.
2014. <em>Bayesian Data Analysis, 3rd Edition</em>. CRC Press. New
York.</p>
<p>Johnson, N.L., Kotz, S., and Balakrishnan, N. 1997. Discrete
multivariate distributions. Wiley &amp; Sons, New York.</p>
<p>Lee. E., T. Dann, and H. Hoyt. 2021. Yukon River Chinook Genetic
Baseline Improvements. Yukon River Panel Restoration and Enhancement
Fund Final Report, URE-163-19N.</p>
<p>Moran, B.M. and E.C. Anderson. 2018. Bayesian inference from the
conditional genetic stock identification model. <em>Canadian Journal of
Fisheries and Aquatic Sciences</em>. 76(4):551-560. <a href="https://doi.org/10.1139/cjfas-2018-0016" class="external-link uri">https://doi.org/10.1139/cjfas-2018-0016</a></p>
<p>Pella, J. and M. Masuda. 2001. Bayesian methods for analysis of stock
mixtures from genetic characters. <em>Fish. Bull.</em> 99:151–167.</p>
<p>Robert, C. and G. Casella. 2010. <em>Introducing Monte Carlo Methods
with R</em>. Springer. New York.</p>
<p>Templin, W. D., J. E. Seeb, J. R. Jasper, A. W. Barclay, L. W. Seeb.
2011. Genetic differentiation of Alaska Chinook salmon: the missing link
for migratory studies. <em>Mol Ecol Resour</em>. 11(Suppl 1):226-246. <a href="doi:10.1111/j.1755-0998.2010.02968.x" class="uri">doi:10.1111/j.1755-0998.2010.02968.x</a>.</p>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Bobby Hsu.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
