<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Ms.GSI in the multistage of genetic stock identification madness • Ms.GSI</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Ms.GSI in the multistage of genetic stock identification madness">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">Ms.GSI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/msgsi_vignette.html">Ms.GSI in the multistage of genetic stock identification madness</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boppingshoe/Ms.GSI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Ms.GSI in the multistage of genetic stock identification madness</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/boppingshoe/Ms.GSI/blob/master/vignettes/msgsi_vignette.Rmd" class="external-link"><code>vignettes/msgsi_vignette.Rmd</code></a></small>
      <div class="d-none name"><code>msgsi_vignette.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boppingshoe/Ms.GSI" class="external-link">Ms.GSI</a></span><span class="op">)</span></span>
<span><span class="co"># devtools::load_all()</span></span></code></pre></div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>This document contains the background information for a integrated
multistage genetic stock identification (GSI) model in two parts. The
first part the describes how to use package <em>Ms.GSI</em> to conduct a
GSI analysis. The steps include formatting input data, running the
integrated multistage GSI model, and summarizing results. The second
part details the technical background of integrated multistage GSI model
and its mathematical theory. There is a separate article describe the
general background of the integrated multistage framework (<em>add link
to integrated multistage paper</em>).</p>
</div>
<div class="section level2">
<h2 id="how-to-use-ms-gsi">How to use <em>Ms.GSI</em><a class="anchor" aria-label="anchor" href="#how-to-use-ms-gsi"></a>
</h2>
<p><em>Ms.GSI</em> follows the work flow: format input data, run
integrated multistage model, and summarize results/convergence
diagnostics.</p>
<div class="section level3">
<h3 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h3>
<p>There are few pieces of information needed for input data set:</p>
<ul>
<li>broad-scale baseline</li>
<li>regional baseline</li>
<li>mixture sample</li>
<li>broad-scale population information</li>
<li>regional population information</li>
</ul>
<p>There are pre-loaded example data sets available in <em>Ms.GSI</em>.
We will look at them one at a time. Note that the example data sets are
simulated using existing baseline archived by the Alaska Department of
Fish &amp; Game Gene Conservation Lab (GCL). The fabricated data set
does not represent the true population proportions in real fisheries. In
this example, we made up a scenario similar to the Bering Sea groundfish
fisheries where Chinook salmon harvested as bycatch originated from a
wide range of geographic areas. Within the bycatch sample, we were
interested in the proportion contribution from the lower, middle and
upper Yukon River reporting groups (see map below). We used a coast-wide
data set for Chinook salmon (Templin et al. 2011; Templin baseline
hereafter) as the broad-scale baseline to separate Yukon River fish from
other non-Yukon stocks in our data set during the first stage of the
analysis. However, genetic markers in the Templin baseline were unable
to clearly distinguish between lower Yukon River and other coastal
western Alaska populations, so we used a second baseline with additional
genetic markers that were specifically designed to differentiate Yukon
River Chinook salmon populations (Lee et al. 2021; Yukon River baseline
hereafter) as the regional fine-scale baseline for the second stage.</p>
<p>It is important to note that in the original grouping of the Templin
baseline, Lower Yukon was a part of Coastal Western Alaska reporting
group. We isolated Lower Yukon from the rest of the Coastal Western
Alaska, but the accuracy of the proportion estimates would likely
diminish because of the breakup of Coastal Western Alaska reporting
group. We do not recommend using a genetic baseline beyond its original
design. And researchers should be aware of the capability of their
genetic baselines before utilizing them in the integrated multistage
model. In our example, it would be ideal to keep the Coastal Western
Alaska group intact in the broad-scale baseline, and break up the group
into Lower Yukon and others using a fine-scale regional baseline.
However, at the time of writing, such regional baseline with adequate
resolution was still in development.</p>
<div class="float">
<img src="small_map_bering_chinook.png" style="width:100.0%" alt="Collection locations and color-coded reporting groups of Chinook salmon represented in the Yukon River example. Shaded area represents the Yukon River region."><div class="figcaption">Collection locations and color-coded reporting
groups of Chinook salmon represented in the Yukon River example. Shaded
area represents the Yukon River region.</div>
</div>
<p>We assembled a mixture sample containing 150 individuals from
collection sites across Yukon River, coastal western Alaskan, Alaska
Peninsula, Gulf of Alaska, and Kamchatka Peninsula (Russia). The
collections were grouped into five reporting groups: Lower Yukon, Middle
Yukon, Upper Yukon, Coastal Western Alaska (Coastal West Alaska), and
Others.</p>
<div class="section level4">
<h4 id="mixture">Mixture<a class="anchor" aria-label="anchor" href="#mixture"></a>
</h4>
<p>First we will take a look at the baseline and mixture samples.
<em>Ms.GSI</em> accepts the genotype information in two format: 1) GCL
format or 2) package <em>rubias</em> format. The example data sets are
in <em>rubias</em> format and naming convention, but procedures for GCL
format are the same<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;Note for the GCL folks: you don’t need to transform GCL
data sets to &lt;em&gt;rubias&lt;/em&gt; format. &lt;em&gt;Ms.GSI&lt;/em&gt; got you covered!&lt;/p&gt;"><sup>1</sup></a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">mix</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 150 × 358</span></span></span>
<span><span class="co">#&gt;    sample_type repunit collection indiv   `GTH2B-550` `GTH2B-550.1` NOD1  NOD1.1</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_1  C           G             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_2  C           C             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_3  C           C             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_4  C           C             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_5  C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_6  C           C             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_7  C           C             <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_8  C           G             <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_9  C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> mixture     <span style="color: #BB0000;">NA</span>      Bering Sea fish_10 C           G             <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 140 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 350 more variables: `Ots_100884-287` &lt;chr&gt;, `Ots_100884-287.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_101554-407` &lt;chr&gt;, `Ots_101554-407.1` &lt;chr&gt;, `Ots_102414-395` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102414-395.1` &lt;chr&gt;, `Ots_102867-609` &lt;chr&gt;, `Ots_102867-609.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103041-52` &lt;chr&gt;, `Ots_103041-52.1` &lt;chr&gt;, `Ots_103122-180` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103122-180.1` &lt;chr&gt;, `Ots_104048-194` &lt;chr&gt;, `Ots_104048-194.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_104063-132` &lt;chr&gt;, `Ots_104063-132.1` &lt;chr&gt;, `Ots_104415-88` &lt;chr&gt;, …</span></span></span></code></pre></div>
<p>Columns 5 to 358 contain genotype information for loci in
<strong>BOTH</strong> broad-scale and regional baselines. You do not
have to specify the loci for each baseline (you can if you want to
double check, more on that later). <em>Ms.GSI</em> matches the loci
between mixture and baselines as long as the locus names are
consistent.</p>
<p>If you have fish with known-origin, you can specify their identities
by adding a column called <code>known_collection</code> in the mixture
data set. The entry for known-origin should match the collection name in
the broad-scale baseline. Fish with unknown-origin should have a
<code>NA</code> entry.</p>
</div>
<div class="section level4">
<h4 id="broad-scale-baseline">Broad-scale baseline<a class="anchor" aria-label="anchor" href="#broad-scale-baseline"></a>
</h4>
<p>Next, we will take a look at the broad-scale (Templin) baseline
example provided in <em>Ms.GSI</em>. There are originally 45 loci in the
Templin baseline, but we reduced the marker set to 28 loci due to
limitation on the data size (and other technical reasons). However, for
demonstration purpose, this data set will suffice.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">base_templin</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 29,363 × 60</span></span></span>
<span><span class="co">#&gt;    sample_type repunit collection indiv   `GTH2B-550` `GTH2B-550.1` NOD1  NOD1.1</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> reference   Russia  KBIST98L   KBIST9… C           C             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> reference   Russia  KBIST98L   KBIST9… C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> reference   Russia  KBIST98L   KBIST9… C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> reference   Russia  KBIST98L   KBIST9… C           C             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> reference   Russia  KBIST98L   KBIST9… C           G             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> reference   Russia  KBIST98L   KBIST9… C           C             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> reference   Russia  KBIST98L   KBIST9… C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> reference   Russia  KBIST98L   KBIST9… C           C             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> reference   Russia  KBIST98L   KBIST9… C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> reference   Russia  KBIST98L   KBIST9… C           C             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 29,353 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 52 more variables: `Ots_2KER-137` &lt;chr&gt;, `Ots_2KER-137.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_AsnRS-72` &lt;chr&gt;, `Ots_AsnRS-72.1` &lt;chr&gt;, Ots_ETIF1A &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Ots_ETIF1A.1 &lt;chr&gt;, `Ots_GPH-318` &lt;chr&gt;, `Ots_GPH-318.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_GST-207` &lt;chr&gt;, `Ots_GST-207.1` &lt;chr&gt;, `Ots_hnRNPL-533` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_hnRNPL-533.1` &lt;chr&gt;, `Ots_HSP90B-100` &lt;chr&gt;, `Ots_HSP90B-100.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_IGF1-91` &lt;chr&gt;, `Ots_IGF1-91.1` &lt;chr&gt;, `Ots_IK1-328` &lt;chr&gt;, …</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="regional-baseline">Regional baseline<a class="anchor" aria-label="anchor" href="#regional-baseline"></a>
</h4>
<p>The regional baseline (Yukon) is in the same format. There are
originally 380 loci in the Yukon River Chinook baseline, but we reduced
the numbers to 177 in this demonstration.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">base_yukon</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5,435 × 358</span></span></span>
<span><span class="co">#&gt;    sample_type repunit   collection indiv `GTH2B-550` `GTH2B-550.1` NOD1  NOD1.1</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> reference   Lower Yu… KANDR02.K… KAND… G           G             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> reference   Lower Yu… KANDR02.K… KAND… C           G             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> reference   Lower Yu… KANDR02.K… KAND… G           G             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> reference   Lower Yu… KANDR02.K… KAND… G           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> reference   Lower Yu… KANDR02.K… KAND… C           G             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> reference   Lower Yu… KANDR02.K… KAND… C           C             C     C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> reference   Lower Yu… KANDR02.K… KAND… C           G             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> reference   Lower Yu… KANDR02.K… KAND… C           C             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> reference   Lower Yu… KANDR02.K… KAND… C           C             G     G     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> reference   Lower Yu… KANDR02.K… KAND… C           G             C     G     </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 5,425 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 350 more variables: `Ots_100884-287` &lt;chr&gt;, `Ots_100884-287.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_101554-407` &lt;chr&gt;, `Ots_101554-407.1` &lt;chr&gt;, `Ots_102414-395` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102414-395.1` &lt;chr&gt;, `Ots_102867-609` &lt;chr&gt;, `Ots_102867-609.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103041-52` &lt;chr&gt;, `Ots_103041-52.1` &lt;chr&gt;, `Ots_103122-180` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103122-180.1` &lt;chr&gt;, `Ots_104048-194` &lt;chr&gt;, `Ots_104048-194.1` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_104063-132` &lt;chr&gt;, `Ots_104063-132.1` &lt;chr&gt;, `Ots_104415-88` &lt;chr&gt;, …</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="population-information">Population information<a class="anchor" aria-label="anchor" href="#population-information"></a>
</h4>
<p>Another piece of information needed is the population details for
each baseline. You need to include three columns in the population
information table. Column <code>collection</code> contains the names for
each population in the baseline. Column <code>reunit</code> specifies
the reporting group each population belongs to. Column
<code>grpvec</code> specifies the identification number for each
reporting group. Below shows the first ten rows of population
information for the Templin (broad-scale) baseline.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">templin_pops211</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 211 × 3</span></span></span>
<span><span class="co">#&gt;    collection       repunit             grpvec</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> KBIST98L         Russia                   1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> KBOLS02.KBOLS98E Russia                   1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> KKAMC97.KKAMC98L Russia                   1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> KPAKH02          Russia                   1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> KPILG05.KPILG06  Coastal West Alaska      2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> KUNAL05          Coastal West Alaska      2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> KGOLS05.KGOLS06  Coastal West Alaska      2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> KANDR02.KANDR03  Lower Yukon              3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> KANVI02          Lower Yukon              3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> KGISA01          Lower Yukon              3</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 201 more rows</span></span></span></code></pre></div>
<p>If you have hatchery populations in your mixture sample, you can tell
<em>Ms.GSI</em> either a collection belongs to natural or
hatchery-origin by adding an <code>origin</code> column in the
population information table. In the <code>origin</code> column, you
identify each collection with <code>"wild"</code> or
<code>"hatchery"</code>. If you don’t care to separate natural and
hatchery origins, you can lump them in one collection. In this case, you
don’t need to add an <code>origin</code> column. Also, this option to
identify hatchery fish is only available in the broad-scale baseline, so
don’t add an <code>origin</code> column in the population table for
regional baseline.</p>
<p>Population information table for the Yukon (regional) baseline is in
the same format, but not necessarily in the same order.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">yukon_pops50</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 50 × 3</span></span></span>
<span><span class="co">#&gt;    collection                          grpvec repunit     </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> KANDR02.KANDR03                          1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> KANVI03.KANVI07                          1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> KNUL12SF                                 1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> KNUL12NF                                 1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> KGISA01                                  1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> KKATE02.KKATE12                          1 Lower Yukon </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> KHENS01                                  2 Middle Yukon</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> KHENS07.KHENS15                          2 Middle Yukon</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> KSFKOY03                                 2 Middle Yukon</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> KMFKOY10.KMFKOY11.KMFKOY12.KMFKOY13      2 Middle Yukon</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 40 more rows</span></span></span></code></pre></div>
<p>Once you have the data files ready (I recommend saving them as .Rdata
or .Rds files), you can use <code><a href="../reference/prep_msgsi_data.html">prep_msgsi_data()</a></code> function to
convert them into input data set for model run. You’ll also need to
identify “groups of interest” in parameter <code>sub-group</code>. In
this example, groups of interests are Lower Yukon, Middle Yukon and
Upper Yukon reporting groups. Their identify numbers are 3, 4, and 5 in
the broad-scale baseline. There’s an option to save the input data at a
designated directory by identifying the location in parameter
<code>file_path</code>.</p>
<p><code><a href="../reference/prep_msgsi_data.html">prep_msgsi_data()</a></code> function matches the loci between
mixture and baselines. But if you want to make sure that you didn’t miss
any locus in your baselines or mixture, you can manually provide loci
names (in string vector) for each baseline by inputting them in
<code>loci1</code> and <code>loci2</code>. In this example we don’t
manually provide lists of loci because we trust that mixture and
baselines all have the correct loci.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">msgsi_dat</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/prep_msgsi_data.html">prep_msgsi_data</a></span><span class="op">(</span>mixture_data <span class="op">=</span> <span class="va">mix</span>,</span>
<span>  baseline1_data <span class="op">=</span> <span class="va">base_templin</span>, baseline2_data <span class="op">=</span> <span class="va">base_yukon</span>,</span>
<span>  pop1_info <span class="op">=</span> <span class="va">templin_pops211</span>, pop2_info <span class="op">=</span> <span class="va">yukon_pops50</span>, sub_group <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Compiling input data, may take a minute or two...</span></span>
<span><span class="co">#&gt; Time difference of 8.876829 secs</span></span></code></pre></div>
<p><code><a href="../reference/prep_msgsi_data.html">prep_msgsi_data()</a></code> formats the data files and put them in
a list. It took few seconds to format the input data in this case.
Bigger data sets may take longer. Here are the first few rows/items in
the input data list:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">msgsi_dat</span>, <span class="va">head</span><span class="op">)</span></span>
<span><span class="co">#&gt; $x</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 57</span></span></span>
<span><span class="co">#&gt;   indiv    `GTH2B-550_1` `GTH2B-550_2` NOD1_1 NOD1_2 `Ots_2KER-137_1`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> fish_1               1             1      2      0                0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> fish_10              1             1      0      0                1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> fish_100             1             1      1      1                1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> fish_101             1             1      1      1                1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> fish_102             0             2      0      2                0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> fish_103             2             0      1      1                1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 51 more variables: `Ots_2KER-137_2` &lt;int&gt;, `Ots_AsnRS-72_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_AsnRS-72_2` &lt;int&gt;, Ots_ETIF1A_1 &lt;int&gt;, Ots_ETIF1A_2 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_GPH-318_1` &lt;int&gt;, `Ots_GPH-318_2` &lt;int&gt;, `Ots_GST-207_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_GST-207_2` &lt;int&gt;, `Ots_HSP90B-100_1` &lt;int&gt;, `Ots_HSP90B-100_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_IGF1-91_1` &lt;int&gt;, `Ots_IGF1-91_2` &lt;int&gt;, `Ots_IK1-328_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_IK1-328_2` &lt;int&gt;, `Ots_LEI-292_1` &lt;int&gt;, `Ots_LEI-292_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Ots_MHC1_1 &lt;int&gt;, Ots_MHC1_2 &lt;int&gt;, `Ots_OPLW-173_1` &lt;int&gt;, …</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $x2</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 355</span></span></span>
<span><span class="co">#&gt;   indiv    `GTH2B-550_1` `GTH2B-550_2` NOD1_1 NOD1_2 `Ots_100884-287_1`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> fish_1               1             1      2      0                  2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> fish_10              1             1      0      0                  0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> fish_100             1             1      1      1                  1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> fish_101             1             1      1      1                  2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> fish_102             2             0      0      2                  1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> fish_103             0             2      1      1                  1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 349 more variables: `Ots_100884-287_2` &lt;int&gt;, `Ots_101554-407_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_101554-407_2` &lt;int&gt;, `Ots_102414-395_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102414-395_2` &lt;int&gt;, `Ots_102867-609_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102867-609_2` &lt;int&gt;, `Ots_103041-52_1` &lt;int&gt;, `Ots_103041-52_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103122-180_1` &lt;int&gt;, `Ots_103122-180_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_104048-194_1` &lt;int&gt;, `Ots_104048-194_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_104063-132_1` &lt;int&gt;, `Ots_104063-132_2` &lt;int&gt;, …</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $y</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 59</span></span></span>
<span><span class="co">#&gt;   collection            repunit grpvec `GTH2B-550_1` `GTH2B-550_2` NOD1_1 NOD1_2</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CHBIG92.KIBIG93.KBIG… Northe…      9           254            82    104    228</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> CHCRY92.KICRY94.KCRY… Coasta…     10           306           302    120    488</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> CHDMT92.KDEER94       Coasta…     10           178           116     77    217</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> CHKAN92.KIKAN93.KKAN… Coasta…      2           341           147    281    199</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> CHKOG92.KIKOG93.KKOG… Coasta…      2           205            91    191    105</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> CHNUU92.KINUS93       Coasta…      2            85            27     73     41</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 52 more variables: `Ots_2KER-137_1` &lt;int&gt;, `Ots_2KER-137_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_AsnRS-72_1` &lt;int&gt;, `Ots_AsnRS-72_2` &lt;int&gt;, Ots_ETIF1A_1 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Ots_ETIF1A_2 &lt;int&gt;, `Ots_GPH-318_1` &lt;int&gt;, `Ots_GPH-318_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_GST-207_1` &lt;int&gt;, `Ots_GST-207_2` &lt;int&gt;, `Ots_HSP90B-100_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_HSP90B-100_2` &lt;int&gt;, `Ots_IGF1-91_1` &lt;int&gt;, `Ots_IGF1-91_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_IK1-328_1` &lt;int&gt;, `Ots_IK1-328_2` &lt;int&gt;, `Ots_LEI-292_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_LEI-292_2` &lt;int&gt;, Ots_MHC1_1 &lt;int&gt;, Ots_MHC1_2 &lt;int&gt;, …</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $y2</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 357</span></span></span>
<span><span class="co">#&gt;   collection            grpvec repunit `GTH2B-550_1` `GTH2B-550_2` NOD1_1 NOD1_2</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CHSID92j                   3 Upper …             7           183    116     74</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> K100MILECR16.K100MIL…      3 Upper …             7           103     78     34</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> KANDR02.KANDR03            1 Lower …            78           230    208    100</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> KANVI03.KANVI07            1 Lower …            62           164    131     79</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> KBEAV97                    2 Middle…            40           148    152     38</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> KBIGS87.KBIGS07            3 Upper …            30           258    231     65</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 350 more variables: `Ots_100884-287_1` &lt;int&gt;, `Ots_100884-287_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_101554-407_1` &lt;int&gt;, `Ots_101554-407_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102414-395_1` &lt;int&gt;, `Ots_102414-395_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_102867-609_1` &lt;int&gt;, `Ots_102867-609_2` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103041-52_1` &lt;int&gt;, `Ots_103041-52_2` &lt;int&gt;, `Ots_103122-180_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_103122-180_2` &lt;int&gt;, `Ots_104048-194_1` &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Ots_104048-194_2` &lt;int&gt;, `Ots_104063-132_1` &lt;int&gt;, …</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $iden</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $nalleles</span></span>
<span><span class="co">#&gt;    GTH2B-550         NOD1 Ots_2KER-137 Ots_AsnRS-72   Ots_ETIF1A  Ots_GPH-318 </span></span>
<span><span class="co">#&gt;            2            2            2            2            2            2 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $nalleles2</span></span>
<span><span class="co">#&gt;      GTH2B-550           NOD1 Ots_100884-287 Ots_101554-407 Ots_102414-395 </span></span>
<span><span class="co">#&gt;              2              2              2              2              2 </span></span>
<span><span class="co">#&gt; Ots_102867-609 </span></span>
<span><span class="co">#&gt;              2 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $groups</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 3</span></span></span>
<span><span class="co">#&gt;   collection                      repunit                  grpvec</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CHBIG92.KIBIG93.KBIGB04.KBIGB95 Northeast Gulf of Alaska      9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> CHCRY92.KICRY94.KCRYA05         Coastal Southeast Alaska     10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> CHDMT92.KDEER94                 Coastal Southeast Alaska     10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> CHKAN92.KIKAN93.KKANE05         Coastal West Alaska           2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> CHKOG92.KIKOG93.KKOGR05         Coastal West Alaska           2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> CHNUU92.KINUS93                 Coastal West Alaska           2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $p2_groups</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 3</span></span></span>
<span><span class="co">#&gt;   collection                repunit      grpvec</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CHSID92j                  Upper Yukon       3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> K100MILECR16.K100MILECR15 Upper Yukon       3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> KANDR02.KANDR03           Lower Yukon       1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> KANVI03.KANVI07           Lower Yukon       1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> KBEAV97                   Middle Yukon      2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> KBIGS87.KBIGS07           Upper Yukon       3</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $comb_groups</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 3</span></span></span>
<span><span class="co">#&gt;   collection                      repunit                  grpvec</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CHBIG92.KIBIG93.KBIGB04.KBIGB95 Northeast Gulf of Alaska      9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> CHCRY92.KICRY94.KCRYA05         Coastal Southeast Alaska     10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> CHDMT92.KDEER94                 Coastal Southeast Alaska     10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> CHKAN92.KIKAN93.KKANE05         Coastal West Alaska           2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> CHKOG92.KIKOG93.KKOGR05         Coastal West Alaska           2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> CHNUU92.KINUS93                 Coastal West Alaska           2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $sub_group</span></span>
<span><span class="co">#&gt; [1] 3 4 5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $group_names_t1</span></span>
<span><span class="co">#&gt; [1] "Russia"                 "Coastal West Alaska"    "Lower Yukon"           </span></span>
<span><span class="co">#&gt; [4] "Middle Yukon"           "Upper Yukon"            "North Alaska Peninsula"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $group_names_t2</span></span>
<span><span class="co">#&gt; [1] "Lower Yukon"  "Middle Yukon" "Upper Yukon" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $wildpops</span></span>
<span><span class="co">#&gt; [1] "CHBIG92.KIBIG93.KBIGB04.KBIGB95" "CHCRY92.KICRY94.KCRYA05"        </span></span>
<span><span class="co">#&gt; [3] "CHDMT92.KDEER94"                 "CHKAN92.KIKAN93.KKANE05"        </span></span>
<span><span class="co">#&gt; [5] "CHKOG92.KIKOG93.KKOGR05"         "CHNUU92.KINUS93"                </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $hatcheries</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="genetic-stock-identification">Genetic stock identification<a class="anchor" aria-label="anchor" href="#genetic-stock-identification"></a>
</h3>
<p>Once your input data set is ready, you can use
<code><a href="../reference/msgsi_mdl.html">msgsi_mdl()</a></code> to run the model. If you are used to running
<em>rubias</em>, <em>Ms.GSI</em> might feel a bit slow. That is because
1) we are running two GSI models in tandem, so it takes twice as long
than running a single model, and 2) <em>Ms.GSI</em> is written solely in
<em>R</em>, which is not as computationally efficient as language
<em>C</em>. So, why not code <em>Ms.GSI</em> in <em>C</em>? Because
we’re not technically advanced like the folks who developed
<em>rubias</em> package (i.e., we don’t know how to code in
<em>C++</em>).</p>
<p>Because of the running time, we recommend running the integrated
multistage model in conditional GSI mode (default setting). But there is
an option to run the model in fully Bayesian mode if one choose to. If
you run the model in fully Bayesian mode, you have the option to include
numbers of adaptation run. Some people think that adaptation run
encourages convergence in fully Bayesian mode. We have not test that
theory but provide the option for those who want to try it.</p>
<p>We demonstrate the model run with one chains of 150 iterations (first
50 as warm-up runs, or burn-ins). We only run one chain in this example
so it can pass CMD check while building the vignette document<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;The package we use to build &lt;em&gt;Ms.GSI&lt;/em&gt; has an
automated checking process. For some reason, automated check would fail
if we ran multiple chains in our examples. Probably something to do with
the parallel process we included in our function.&lt;/p&gt;"><sup>2</sup></a>. In
reality, you should <strong>definitely</strong> run multiple chains with
more iterations. There are also options to keep the burn-ins and set
random seed for reproducible results. We don’t show them in this example
though (but you can always <code><a href="../reference/msgsi_mdl.html">?msgsi_mdl</a></code>).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">msgsi_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/msgsi_mdl.html">msgsi_mdl</a></span><span class="op">(</span><span class="va">msgsi_dat</span>, nreps <span class="op">=</span> <span class="fl">150</span>, nburn <span class="op">=</span> <span class="fl">50</span>, thin <span class="op">=</span> <span class="fl">1</span>, nchains <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Running model (and the category is... Linen Vs. Silk!)</span></span>
<span><span class="co">#&gt; Time difference of 2.161755 secs</span></span>
<span><span class="co">#&gt; October-09-2025 19:38</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summarizing-results">Summarizing results<a class="anchor" aria-label="anchor" href="#summarizing-results"></a>
</h3>
<div class="section level4">
<h4 id="stock-proportions">Stock proportions<a class="anchor" aria-label="anchor" href="#stock-proportions"></a>
</h4>
<p>The output of model contains nine items: <code>summ_t1</code>,
<code>trace_t1</code>, <code>summ_t2</code>, <code>trace_t2</code>,
<code>summ_comb</code>, <code>trace_comb</code>,
<code>comb_groups</code>, <code>comb_groups</code>, <code>iden_t1</code>
and <code>idens_t2</code>. Items with “summ” are summary for reporting
group proportions and associated convergence diagnostics. If you want to
see summaries for stage one and two individually, <code>summ_t1</code>
and <code>summ_t2</code> will show you that. Most people probably want
to see the combined summary, <code>summ_comb</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">msgsi_out</span><span class="op">$</span><span class="va">summ_comb</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 9</span></span></span>
<span><span class="co">#&gt;    group                mean  median      sd    ci.05   ci.95    p0 GR     n_eff</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Russia            5.21<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 4.82<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 0.020<span style="text-decoration: underline;">3</span>  2.29<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 0.085<span style="text-decoration: underline;">7</span>   0    <span style="color: #BB0000;">NA</span>     45.4 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Coastal West Ala… 7.27<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 7.15<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 0.056<span style="text-decoration: underline;">0</span>  2.55<span style="color: #949494;">e</span><span style="color: #BB0000;">-12</span> 0.179    0.12 <span style="color: #BB0000;">NA</span>      7.96</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> North Alaska Pen… 3.59<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 3.45<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 0.019<span style="text-decoration: underline;">7</span>  1.11<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 0.072<span style="text-decoration: underline;">0</span>   0    <span style="color: #BB0000;">NA</span>     29.1 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Northwest Gulf o… 3.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 3.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 0.059<span style="text-decoration: underline;">3</span>  2.14<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 0.410    0    <span style="color: #BB0000;">NA</span>     23.2 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Copper            1.50<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 6.36<span style="color: #949494;">e</span><span style="color: #BB0000;">-6</span> 0.003<span style="text-decoration: underline;">98</span> 3.16<span style="color: #949494;">e</span><span style="color: #BB0000;">-21</span> 0.008<span style="text-decoration: underline;">56</span>  0.39 <span style="color: #BB0000;">NA</span>     67.8 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Northeast Gulf o… 1.38<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 6.74<span style="color: #949494;">e</span><span style="color: #BB0000;">-7</span> 0.004<span style="text-decoration: underline;">24</span> 1.53<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span> 0.007<span style="text-decoration: underline;">83</span>  0.48 <span style="color: #BB0000;">NA</span>     59.2 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Coastal Southeas… 6.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 1.70<span style="color: #949494;">e</span><span style="color: #BB0000;">-6</span> 0.002<span style="text-decoration: underline;">02</span> 4.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span> 0.002<span style="text-decoration: underline;">96</span>  0.44 <span style="color: #BB0000;">NA</span>    100   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> British Columbia  5.57<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 3.83<span style="color: #949494;">e</span><span style="color: #BB0000;">-7</span> 0.002<span style="text-decoration: underline;">14</span> 2.25<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span> 0.002<span style="text-decoration: underline;">45</span>  0.5  <span style="color: #BB0000;">NA</span>    100   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> WA/OR/CA          6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 1.32<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> 0.001<span style="text-decoration: underline;">61</span> 2.55<span style="color: #949494;">e</span><span style="color: #BB0000;">-17</span> 0.004<span style="text-decoration: underline;">13</span>  0.38 <span style="color: #BB0000;">NA</span>    100   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Lower Yukon       2.56<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 2.60<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 0.064<span style="text-decoration: underline;">5</span>  1.56<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 0.358    0    <span style="color: #BB0000;">NA</span>     17.2 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> Middle Yukon      7.60<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 7.59<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 0.022<span style="text-decoration: underline;">2</span>  4.66<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 0.122    0    <span style="color: #BB0000;">NA</span>    100   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> Upper Yukon       1.93<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 1.88<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 0.032<span style="text-decoration: underline;">1</span>  1.42<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 0.240    0    <span style="color: #BB0000;">NA</span>    100</span></span></code></pre></div>
<p>Most column names are self explanatory, but others might need some
additional descriptions. <code>ci.05</code> and <code>ci.95</code> are
the lower and upper bounds of 90% credible interval. <code>p0</code> is
the probability of an estimate equals zero. We estimate <code>p0</code>
by calculating the portion of posterior samples that is less than
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>5</mn><mo>×</mo><msup><mn>10</mn><mrow><mo>−</mo><mn>7</mn></mrow></msup></mrow><annotation encoding="application/x-tex">5 \times 10^{-7}</annotation></semantics></math>.
<code>GR</code> is the Gelman-Rubin statistic (a.k.a.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mover><mi>R</mi><mo accent="true">̂</mo></mover><annotation encoding="application/x-tex">\hat R</annotation></semantics></math>).
In this example, Gelman-Rubin statistic is not calculated because we
only run one chain. <code>n_eff</code> is the effective size, or
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>N</mi><mrow><mi>e</mi><mi>f</mi><mi>f</mi></mrow></msub><annotation encoding="application/x-tex">N_{eff}</annotation></semantics></math>.
We will not discuss how to diagnose convergence in this document. Please
consult Gelman et al. 2014, Gelman &amp; Rubin 1992, Brooks &amp; Gelman
1998 and other literature on statistical methods.</p>
<p>Items with “trace” are the posterior sample history, or trace
history, for either stage one, two, or combined. Trace history is needed
for making trace plots. And if you need to combine reporting groups
proportions or combine variance, trace histories are what you need.
<code>trace_</code> items are tibbles with each collection as a column.
There are two additional columns, <code>itr</code> and
<code>chain</code>, to identify Markov chain Monte Carlo (MCMC) sampling
iteration and chain.</p>
<p><code>comb_groups</code> are provided in the output as reference for
functions making trace plots (<code><a href="../reference/tr_plot.html">tr_plot()</a></code>) and stratified
estimator (<code><a href="../reference/stratified_estimator_msgsi.html">stratified_estimator_msgsi()</a></code>). Grouping for
stage one and two can also be found in the input data.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">msgsi_out</span><span class="op">$</span><span class="va">trace_comb</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 238</span></span></span>
<span><span class="co">#&gt;    CHBIG92.KIBIG93.KBIGB04.KBIGB95 CHCRY92.KICRY94.KCRYA05 CHDMT92.KDEER94</span></span>
<span><span class="co">#&gt;                              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>                       2.65<span style="color: #949494;">e</span><span style="color: #BB0000;">-  3</span>               3.74<span style="color: #949494;">e</span><span style="color: #BB0000;">-223</span>       8.74<span style="color: #949494;">e</span><span style="color: #BB0000;">- 41</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>                       1.49<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span>               9.40<span style="color: #949494;">e</span><span style="color: #BB0000;">- 82</span>       1.64<span style="color: #949494;">e</span><span style="color: #BB0000;">- 39</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>                       4.62<span style="color: #949494;">e</span><span style="color: #BB0000;">-  3</span>               1.47<span style="color: #949494;">e</span><span style="color: #BB0000;">-102</span>       9.82<span style="color: #949494;">e</span><span style="color: #BB0000;">- 99</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>                       3.60<span style="color: #949494;">e</span><span style="color: #BB0000;">-125</span>               2.03<span style="color: #949494;">e</span><span style="color: #BB0000;">- 34</span>       2.85<span style="color: #949494;">e</span><span style="color: #BB0000;">-176</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>                       2.80<span style="color: #949494;">e</span><span style="color: #BB0000;">- 38</span>               7.29<span style="color: #949494;">e</span><span style="color: #BB0000;">- 58</span>       2.79<span style="color: #949494;">e</span><span style="color: #BB0000;">- 72</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>                       7.57<span style="color: #949494;">e</span><span style="color: #BB0000;">- 87</span>               9.73<span style="color: #949494;">e</span><span style="color: #BB0000;">-231</span>       3.04<span style="color: #949494;">e</span><span style="color: #BB0000;">- 24</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>                       7.36<span style="color: #949494;">e</span><span style="color: #BB0000;">- 27</span>               7.77<span style="color: #949494;">e</span><span style="color: #BB0000;">- 22</span>       3.52<span style="color: #949494;">e</span><span style="color: #BB0000;">-  8</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>                       1.16<span style="color: #949494;">e</span><span style="color: #BB0000;">- 47</span>               1.75<span style="color: #949494;">e</span><span style="color: #BB0000;">-164</span>       3.04<span style="color: #949494;">e</span><span style="color: #BB0000;">-154</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>                       9.03<span style="color: #949494;">e</span><span style="color: #BB0000;">- 29</span>               1.23<span style="color: #949494;">e</span><span style="color: #BB0000;">-245</span>       2.33<span style="color: #949494;">e</span><span style="color: #BB0000;">- 17</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>                       1.61<span style="color: #949494;">e</span><span style="color: #BB0000;">- 40</span>               4.42<span style="color: #949494;">e</span><span style="color: #BB0000;">-320</span>       2.06<span style="color: #949494;">e</span><span style="color: #BB0000;">- 22</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 90 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 235 more variables: CHKAN92.KIKAN93.KKANE05 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   CHKOG92.KIKOG93.KKOGR05 &lt;dbl&gt;, CHNUU92.KINUS93 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   CHTAH92.KTAHI04 &lt;dbl&gt;, CHWHI92.KWHIT98.KWHITC05 &lt;dbl&gt;, KALSE04 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   KANCH06.KANCH10 &lt;dbl&gt;, KANDR89.KANDR04 &lt;dbl&gt;, KAROL05 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   KBENJ05.KBENJ06 &lt;dbl&gt;, KBIGCK04 &lt;dbl&gt;, KBIGQU96 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   KBIRK01.KBIRK02.KBIRK03.KBIRK97.KBIRK99 &lt;dbl&gt;, KBIST98L &lt;dbl&gt;, …</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="individual-assignments">Individual assignments<a class="anchor" aria-label="anchor" href="#individual-assignments"></a>
</h4>
<p>The last two items in the output are the identity assignment history
of each individual in the mixture sample. Each column represents an
individual in the mixture, and each row records the identity assigned
during each iteration in each chain. These numbers are the population
identifiers in the same order as your population information files (and
baseline files). Individuals are ordered in the same as the input data
(i.e., mixture data).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">msgsi_out</span><span class="op">$</span><span class="va">idens_t1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 150</span></span></span>
<span><span class="co">#&gt;       V1    V2    V3    V4    V5    V6    V7    V8    V9   V10   V11   V12   V13</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>   200   200    11    97    97   125    97   101   101   114    11   157    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>   200    32    21    87   114    11    97    32   200   114    97   200   200</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>   200    11    21    87   101   125    97    87    97   114    11   200    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>    11   200    21    87   101    11    97   200    11   114    11   200    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>    11    11    21    32   101   125    97   200    97   101    11    11    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>    87    97    21    87    32   125    97    32   101    87    11    32   200</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>    11    87    21    11   101   125    97    32   200   114    11   157    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>    32    21    21    87    87   125    97    87    97   114    97    11    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>    87    87    21    87   200   125    97    32    32   114    87   101    11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>   200    97    21   101    97   200    97    87    32    87    11    97   200</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 90 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 137 more variables: V14 &lt;int&gt;, V15 &lt;int&gt;, V16 &lt;int&gt;, V17 &lt;int&gt;, V18 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   V19 &lt;int&gt;, V20 &lt;int&gt;, V21 &lt;int&gt;, V22 &lt;int&gt;, V23 &lt;int&gt;, V24 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   V25 &lt;int&gt;, V26 &lt;int&gt;, V27 &lt;int&gt;, V28 &lt;int&gt;, V29 &lt;int&gt;, V30 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   V31 &lt;int&gt;, V32 &lt;int&gt;, V33 &lt;int&gt;, V34 &lt;int&gt;, V35 &lt;int&gt;, V36 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   V37 &lt;int&gt;, V38 &lt;int&gt;, V39 &lt;int&gt;, V40 &lt;int&gt;, V41 &lt;int&gt;, V42 &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   V43 &lt;int&gt;, V44 &lt;int&gt;, V45 &lt;int&gt;, V46 &lt;int&gt;, V47 &lt;int&gt;, V48 &lt;int&gt;, …</span></span></span></code></pre></div>
<p>Individual identity output in this format may not be very useful for
most users. So, <em>Ms.GSI</em> has a function
<code><a href="../reference/indiv_assign.html">indiv_assign()</a></code> that summarize the reporting group
assignment probabilities for each individual in the mixture.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/indiv_assign.html">indiv_assign</a></span><span class="op">(</span>mdl_out <span class="op">=</span> <span class="va">msgsi_out</span>, mdl_dat <span class="op">=</span> <span class="va">msgsi_dat</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 150 × 13</span></span></span>
<span><span class="co">#&gt;    ID       Russia `Coastal West Alaska` `North Alaska Peninsula`</span></span>
<span><span class="co">#&gt;  <span style="color: #BCBCBC;">*</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> fish_1     0                     0.31                     0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> fish_10    0.11                  0.07                     0.05</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> fish_100   0.96                  0                        0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> fish_101   0                     0.05                     0.01</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> fish_102   0                     0.03                     0.06</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> fish_103   0                     0.01                     0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> fish_104   0                     0                        0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> fish_105   0                     0.1                      0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> fish_106   0                     0.15                     0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> fish_107   0                     0                        0.62</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 140 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 9 more variables: `Northwest Gulf of Alaska` &lt;dbl&gt;, Copper &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Northeast Gulf of Alaska` &lt;dbl&gt;, `Coastal Southeast Alaska` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `British Columbia` &lt;dbl&gt;, `WA/OR/CA` &lt;dbl&gt;, `Lower Yukon` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Middle Yukon` &lt;dbl&gt;, `Upper Yukon` &lt;dbl&gt;</span></span></span></code></pre></div>
<p>The summary for individual assignment has a column named
<code>ID</code> that identifies each individual in the mixture. The
reporting groups represent the combined grouping of the broad and
regional baselines. Probabilities in each row should sum up to one.</p>
</div>
<div class="section level4">
<h4 id="trace-plot">Trace plot<a class="anchor" aria-label="anchor" href="#trace-plot"></a>
</h4>
<p><em>Ms.GSI</em> has a function for you to make trace plot and examine
the mixing of MCMC chains. Don’t forget to include the group information
(as in <code>groups</code>, <code>p2_groups</code>, or
<code>comb_groups</code>) for the trace history that you want to
plot.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/tr_plot.html">tr_plot</a></span><span class="op">(</span>obj <span class="op">=</span> <span class="va">msgsi_out</span><span class="op">$</span><span class="va">trace_comb</span>, pop_info <span class="op">=</span> <span class="va">msgsi_out</span><span class="op">$</span><span class="va">comb_groups</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="msgsi_vignette_files/figure-html/unnamed-chunk-14-1.png" alt="Trace plots for reporting group proportions." width="100%"><p class="caption">
Trace plots for reporting group proportions.
</p>
</div>
</div>
<div class="section level4">
<h4 id="stratified-estimator">Stratified estimator<a class="anchor" aria-label="anchor" href="#stratified-estimator"></a>
</h4>
<p><code>stratified_estimator-msgsi()</code> function combines the stock
group estimates of multiple mixtures (i.e., strata) weighted by harvest
numbers or fishing efforts. It is designed to work in the same way as
the <code>stratified_estimator_rubias()</code> function from the
<em>GCLr</em> package by the GCL, although the input format is slightly
different. To do so, you would save the <em>Ms.GSI</em> output for each
mixture in a separate folder under your preferred directory, then
specify the path by
<code>stratified_estimator_rubias(path = ...)</code>.</p>
<p><code>stratified_estimator-msgsi()</code> can also combine variances
of different sources for a single mixture. That is, if you want to
convert your total harvest number (with CV provided) to harvest for
individual reporting groups, <code>stratified_estimator-msgsi()</code>
can do that. In this case, we use the example model output with a new
grouping:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">new_groups</span> <span class="op">&lt;-</span> <span class="va">msgsi_out</span><span class="op">$</span><span class="va">summ_comb</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">group</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>new_repunit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Broad"</span>, <span class="fl">9</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Regional"</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>repunit <span class="op">=</span> <span class="va">group</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/stratified_estimator_msgsi.html">stratified_estimator_msgsi</a></span><span class="op">(</span>mdl_out <span class="op">=</span> <span class="va">msgsi_out</span>, mixvec <span class="op">=</span> <span class="st">"Bering example"</span>,</span>
<span>                           catchvec <span class="op">=</span> <span class="fl">500</span>, cv <span class="op">=</span> <span class="fl">0.1</span>, new_pop_info <span class="op">=</span> <span class="va">new_groups</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 12</span></span></span>
<span><span class="co">#&gt;   repunit  mean_harv sd_harv median_harv ci05_harv ci95_harv  mean     sd median</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Broad         235.    42.5        231.      175.      311. 0.475 0.064<span style="text-decoration: underline;">5</span>  0.475</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Regional      258.    38.7        253.      202.      327. 0.525 0.064<span style="text-decoration: underline;">5</span>  0.525</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 more variables: ci05 &lt;dbl&gt;, ci95 &lt;dbl&gt;, `P=0` &lt;dbl&gt;</span></span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="methods-math">Methods (math!)<a class="anchor" aria-label="anchor" href="#methods-math"></a>
</h2>
<div class="section level3">
<h3 id="pella-masuda-model">Pella-Masuda Model<a class="anchor" aria-label="anchor" href="#pella-masuda-model"></a>
</h3>
<p>This integrated multistage GSI model is essentially two Bayesian GSI
models stacked on top of each other; hence the name “multistage.” The
Pella-Masuda model (Pella &amp; Masuda 2001) is the Bayesian GSI model
that make up each stage in the integrated multistage model. We will
first describe the Pella-Masuda model before discussing the development
of a integrated multistage model.</p>
<p>In a group of mixed populations, Pella-Masuda model assigns
population identities to each individual based on its genetic make-up
(e.g. genotype). Then the model estimates the overall population
proportions based on the numbers of individuals assigned to each
population. In the fishery context, genetic data of the individuals is
called the mixture sample because it consists multi-locus genotype of
individual fish collected from a mixed-stock fishery.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐱</mi><annotation encoding="application/x-tex">\mathbf x</annotation></semantics></math>
denotes the mixture sample. In this document, a bold-font letter
represents a <em>number set</em>, or a collection of distinct elements.
For example,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐱</mi><annotation encoding="application/x-tex">\mathbf x</annotation></semantics></math>
is a set that contains individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>x</mi><annotation encoding="application/x-tex">x</annotation></semantics></math>
elements. And
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><annotation encoding="application/x-tex">x_{m,l,j}</annotation></semantics></math>
is the count of allele
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>
in locus
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>l</mi><annotation encoding="application/x-tex">l</annotation></semantics></math>
for individual fish
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>m</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>M</mi><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">m \in \{1,2,...,M\}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>l</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>L</mi><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">l \in \{1,2,...,L\}</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>j</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>J</mi><mi>l</mi></msub><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">j \in \{1,2,...,J_l\}</annotation></semantics></math>
depends on locus
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>l</mi><annotation encoding="application/x-tex">l</annotation></semantics></math>.</p>
<p>Genetic data of the populations is called the baseline sample because
it consists genotype compositions of various baseline populations
collected at their spawning locations. Researchers select sampling
locations to best represent the populations in an area.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐲</mi><annotation encoding="application/x-tex">\mathbf y</annotation></semantics></math>
denotes the baseline sample.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><annotation encoding="application/x-tex">y_{k,l,j}</annotation></semantics></math>
is the count of allele
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>
in locus
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>l</mi><annotation encoding="application/x-tex">l</annotation></semantics></math>
for a sample of size
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>n</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi></mrow></msub><annotation encoding="application/x-tex">n_{k,l}</annotation></semantics></math>
collected from baseline population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>K</mi><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">k \in \{1,2,...,K\}</annotation></semantics></math>.</p>
<p>For both mixture and baseline samples, it is assumed that allele
counts in each locus follow a multinomial distribution<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;What is a multinomial distribution? They are the counts
of multiple categories based on their corresponding probabilities. For
example, we randomly toss 10 Swedish Fish to three kids. Each time we
toss a fish, the probability of kid #1 catching it is 0.3, kid #2, 0.3,
and kid #3, 0.4. The numbers of fish caught by the three kids are
multinomially distributed with probabilities = {0.3, 0.3, 0.4} and a
total size 10.&lt;/p&gt;"><sup>3</sup></a>. Using another made-up
example, in a baseline sample, there are two allele types in locus 1 for
population 2. Counts for the two alleles are
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>y</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>y</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>2</mn></mrow></msub></mrow><annotation encoding="application/x-tex">y_{1,2,1}, y_{1,2,2}</annotation></semantics></math>,
and they follow a multinomial distribution with parameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>2</mn></mrow></msub></mrow><annotation encoding="application/x-tex">q_{1,2,1}, q_{1,2,2}</annotation></semantics></math>
and size
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>n</mi><mrow><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><annotation encoding="application/x-tex">n_{2,1}</annotation></semantics></math>.
Note that
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>2</mn></mrow></msub></mrow><annotation encoding="application/x-tex">q_{1,2,1}, q_{1,2,2}</annotation></semantics></math>
are the relative frequencies of the two alleles in locus 1 for
population 2. In a Bayesian framework, we need to specify prior
distributions for parameters; therefore, we place a Dirichlet<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;What is a Dirichlet distribution? Using the same example
for multinomial distribution, the fish catching probabilities of all
three kids are Dirichlet-distributed with fish counts of all three kids
as parameters. It is worth noting that the parameters do not have to be
integers as long as they are &amp;gt; 0.&lt;/p&gt;"><sup>4</sup></a> prior
distribution on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>2</mn></mrow></msub></mrow><annotation encoding="application/x-tex">q_{1,2,1}, q_{1,2,2}</annotation></semantics></math>
with hyperparameters<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;A hyperparameter is a parameter of a prior
distribution.&lt;/p&gt;"><sup>5</sup></a>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>β</mi><mrow><mn>1</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>β</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn></mrow></msub></mrow><annotation encoding="application/x-tex">\beta_{1,1}, \beta_{1,2}</annotation></semantics></math>.
Usually we set the priors to be equal for all loci. In this example, let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>β</mi><mrow><mn>1</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>=</mo><msub><mi>β</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn></mrow></msub><mo>=</mo><mn>1</mn><mi>/</mi><mn>2</mn></mrow><annotation encoding="application/x-tex">\beta_{1,1} = \beta_{1,2} = 1/2</annotation></semantics></math>
based on the number of alleles for locus 1.</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
represents
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><annotation encoding="application/x-tex">q_{1,2,1}</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>2</mn></mrow></msub><annotation encoding="application/x-tex">q_{1,2,2}</annotation></semantics></math>,
together with allele frequencies of other loci and other populations. As
you can see,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐲</mi><annotation encoding="application/x-tex">\mathbf y</annotation></semantics></math>
have the same dimension because each relative frequency corresponds to
an allele count. In the model, allele frequencies of baseline
populations,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>,
determine population proportions. And population proportions determines
the identities of individual fish. Individual identities are then
tallied and summarized to update baseline allele frequencies.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐲</mi><annotation encoding="application/x-tex">\mathbf y</annotation></semantics></math>
can be expressed as follows:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐲</mi><mi>k</mi></msub><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐧</mi><mi>k</mi></msub><mo>,</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf y_k \sim Mult(\mathbf n_k, \mathbf q_k)</annotation></semantics></math></p>
<p>Prior distribution for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐪</mi><mi>k</mi></msub><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf q_k \sim Dirich(\mathbf \beta)</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝛃</mi><mo>=</mo><mn>1</mn><mi>/</mi><msub><mi>J</mi><mi>l</mi></msub></mrow><annotation encoding="application/x-tex">\mathbf \beta = 1/J_l</annotation></semantics></math></p>
<p>For mixture sample, allele counts in each locus of individual fish
also follows a multinomial distributions. If a fish came from a certain
population, its distribution of allele counts should resemble the allele
frequencies of the baseline population which it came from. However, the
identity of the individual fish is unknown so it needs to be estimated.
Here we let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐳</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf z_m</annotation></semantics></math>
represent the population identify for the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math><sup>th</sup>
mixture individual.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐳</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf z_m</annotation></semantics></math>
is composed of 0’s and an 1 with a length
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>K</mi><annotation encoding="application/x-tex">K</annotation></semantics></math>
(e.g. number of baseline populations).
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">z_{m,k} = 1</annotation></semantics></math>
if individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belongs to population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>=</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">z_{m,k} = 0</annotation></semantics></math>
otherwise. In a made-up example,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐳</mi><mn>100</mn></msub><mo>=</mo><mo stretchy="false" form="prefix">{</mo><mn>0</mn><mo>,</mo><mn>0</mn><mo>,</mo><mn>1</mn><mo>,</mo><mn>0</mn><mo>,</mo><mn>0</mn><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">\mathbf z_{100} = \{0, 0, 1, 0, 0\}</annotation></semantics></math>
means that there are five baseline populations, and individual fish #100
comes from population 3.</p>
<p>We place a multinomial prior on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mn>2</mn></mrow></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>K</mi></mrow></msub></mrow><annotation encoding="application/x-tex">z_{m,1}, z_{m,2}, ..., z_{m,K}</annotation></semantics></math>
with size 1 and probabilities equal to population proportions
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>1</mn></msub><mo>,</mo><msub><mi>p</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>p</mi><mi>K</mi></msub></mrow><annotation encoding="application/x-tex">p_1, p_2, ..., p_K</annotation></semantics></math>.
We specify a Dirichlet prior distribution on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>1</mn></msub><mo>,</mo><msub><mi>p</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>p</mi><mi>K</mi></msub></mrow><annotation encoding="application/x-tex">p_1, p_2, ..., p_K</annotation></semantics></math>
with hyperparameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>α</mi><mn>1</mn></msub><mo>,</mo><msub><mi>α</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>α</mi><mi>K</mi></msub></mrow><annotation encoding="application/x-tex">\alpha_1, \alpha_2, ..., \alpha_K</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>α</mi><mn>1</mn></msub><mo>=</mo><msub><mi>α</mi><mn>2</mn></msub><mo>=</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>=</mo><msub><mi>α</mi><mi>K</mi></msub><mo>=</mo><mn>1</mn><mi>/</mi><mi>K</mi></mrow><annotation encoding="application/x-tex">\alpha_1 = \alpha_2 = ... = \alpha_K = 1/K</annotation></semantics></math>.
We usually set
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>
to be equal for all reporting groups, but they can be set based on prior
knowledge in population proportions. We express
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐳</mi><annotation encoding="application/x-tex">\mathbf z</annotation></semantics></math>
as follows:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐳</mi><mi>m</mi></msub><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>𝟏</mn><mo>,</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf z_m \sim Mult(\mathbf 1, \mathbf p)</annotation></semantics></math></p>
<p>Prior distribution for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐩</mi><annotation encoding="application/x-tex">\mathbf p</annotation></semantics></math>:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝐩</mi><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf p \sim Dirich(\mathbf \alpha)</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝛂</mi><mo>=</mo><mn>1</mn><mi>/</mi><mi>K</mi></mrow><annotation encoding="application/x-tex">\mathbf \alpha = 1/K</annotation></semantics></math></p>
<p>As mentioned, for mixture sample, allele counts in each locus of
individual fish follows a multinomial distributions. The parameters are
allele frequencies of the corresponding baseline population with size
the numbers of ploidy for each respective locus. Remember that
population identity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">z_{m,k} = 1</annotation></semantics></math>
if individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belongs to population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>=</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">z_{m,k} = 0</annotation></semantics></math>
otherwise. When multiplying population identities,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mn>2</mn></mrow></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>K</mi></mrow></msub></mrow><annotation encoding="application/x-tex">z_{m,1}, z_{m,2}, ..., z_{m,K}</annotation></semantics></math>,
and allele frequencies of baseline populations,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐪</mi><mn>1</mn></msub><mo>,</mo><msub><mi>𝐪</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>𝐪</mi><mi>K</mi></msub></mrow><annotation encoding="application/x-tex">\mathbf q_1, \mathbf q_2, ..., \mathbf q_K</annotation></semantics></math>,
only allele frequencies of baseline population which individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belong to would remain while the rest goes to zero.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐱</mi><annotation encoding="application/x-tex">\mathbf x</annotation></semantics></math>
is expressed below.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mrow><mi>𝐩</mi><mi>𝐥</mi><mi>𝐨</mi><mi>𝐢</mi><mi>𝐝</mi><mi>𝐲</mi></mrow><mo>=</mo><mi>p</mi><mi>l</mi><mi>o</mi><mi>i</mi><mi>d</mi><msub><mi>y</mi><mn>1</mn></msub><mo>,</mo><mi>p</mi><mi>l</mi><mi>o</mi><mi>i</mi><mi>d</mi><msub><mi>y</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>p</mi><mi>l</mi><mi>o</mi><mi>i</mi><mi>d</mi><msub><mi>y</mi><mi>L</mi></msub></mrow><annotation encoding="application/x-tex">\mathbf{ploidy} = ploidy_1, ploidy_2, ..., ploidy_L</annotation></semantics></math>
denotes ploidy for each locus.</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐱</mi><mi>m</mi></msub><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mrow><mi>𝐩</mi><mi>𝐥</mi><mi>𝐨</mi><mi>𝐢</mi><mi>𝐝</mi><mi>𝐲</mi></mrow><mo>,</mo><msub><mi>𝐳</mi><mi>m</mi></msub><mo>⋅</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf x_m \sim Mult(\mathbf{ploidy}, \mathbf z_m \cdot \mathbf q)</annotation></semantics></math></p>
<p>Moran and Anderson (2018) implement a genetic mixture analysis as a
<em>R</em> package, <em>rubias</em>. Their program has been widely used
by researchers around the world, including here at the GCL.
<em>rubias</em> utilizes a model structure called the conditional
genetic stock identification model, or the conditional GSI model, that
is modified from the Pella-Masuda model. The main difference between the
two models is that, in the conditional model,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
is integrated out of the distribution of mixture sample,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐱</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf x_m</annotation></semantics></math>.
That is, baseline allele frequencies are not being updated in the model.
The result of that,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐱</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf x_m</annotation></semantics></math>
takes a form of a compound Dirichlet-multinomial distribution (Johnson
at el. 1997):</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐱</mi><mi>m</mi></msub><mo>∼</mo><mi>C</mi><mi>D</mi><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mrow><mi>𝐩</mi><mi>𝐥</mi><mi>𝐨</mi><mi>𝐢</mi><mi>𝐝</mi><mi>𝐲</mi></mrow><mo>,</mo><msub><mi>𝐳</mi><mi>m</mi></msub><mo>⋅</mo><mi>𝐯</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf x_m \sim CDM(\mathbf{ploidy}, \mathbf z_m \cdot \mathbf v)</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐯</mi><annotation encoding="application/x-tex">\mathbf v</annotation></semantics></math>
is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝛃</mi><mo>+</mo><mi>𝐲</mi></mrow><annotation encoding="application/x-tex">\mathbf \beta + \mathbf y</annotation></semantics></math>.
We are not going to attempt proving the theory behind the conditional
model in this document (details can be found in Moran &amp; Anderson
2018). But since
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
has been integrated out of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐱</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf x_m</annotation></semantics></math>,
the process for estimating parameters is simpler and more streamlined.
We have implemented conditional GSI in each stage of our integrated
multistage model.</p>
</div>
<div class="section level3">
<h3 id="extend-to-multistage">Extend to multistage<a class="anchor" aria-label="anchor" href="#extend-to-multistage"></a>
</h3>
<p>In a multistage setting, we refer to a baseline that covers the whole
range of a mixed stock fishery as a broad-scale baseline. The
broad-scale baseline typically covers a wide range of geographic areas
but does not have a comprehensive collection of reference populations
nor genetic markers to resolve differences between local populations
within a sub-region. These smaller sub-regions of a broad-scale baseline
are covered by regional baselines with higher resolutions. We generalize
the conditions of the <em>Ms.GSI</em> model to allow multiple regional
baselines to be included, although we programmed <em>Ms.GSI</em> to deal
with only one regional baseline vs. one broad-scale baseline.</p>
<p>Let there be
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>B</mi><annotation encoding="application/x-tex">B</annotation></semantics></math>
populations in the broad-scale baseline and indexed as
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>b</mi><mo>=</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>B</mi></mrow><annotation encoding="application/x-tex">b = 1, 2, ..., B</annotation></semantics></math>.
Each of these broad-scale populations may belong to exactly 0 or 1
sub-region for which regional baselines might be available. These
regional baselines have different sets of genetic markers than the
broad-scale baseline and typically include additional populations that
are not represented in the broad-scale baseline. Allow for there to be
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>R</mi><annotation encoding="application/x-tex">R</annotation></semantics></math>
disjoint sub-regions indexed by
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>r</mi><annotation encoding="application/x-tex">r</annotation></semantics></math>,
with each sub-region represented by a distinctive regional baseline. We
employ a superscript
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">^{(r)}</annotation></semantics></math>
upon variables to indicate a quantity associated with regional baseline
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>r</mi><annotation encoding="application/x-tex">r</annotation></semantics></math>.
Populations in different sub-regions cannot overlap, and each population
only occurs once among the regional baselines. Let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
index the populations within these
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>R</mi><annotation encoding="application/x-tex">R</annotation></semantics></math>
regional baselines and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>K</mi><mi>r</mi></msub><annotation encoding="application/x-tex">K_r</annotation></semantics></math>
denotes the number of populations within regional baseline
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>r</mi><annotation encoding="application/x-tex">r</annotation></semantics></math>.</p>
<p>In the <em>Ms.GSI</em> framework, the two stages are connected
because the regional group membership of an individual is conditional on
whether the broad-scale group membership of that individual belongs to
the area of that particular sub-region. The following describes the
conditional relationship between the broad-scale and the regional
baselines:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐳</mi><mi>m</mi></msub><mo>,</mo><msup><mi>ℬ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>=</mo><mrow><mo stretchy="true" form="prefix">{</mo><mtable><mtr><mtd columnalign="left" style="text-align: left"><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup></mtd><mtd columnalign="left" style="text-align: left"><mrow><mtext mathvariant="normal">if </mtext><mspace width="0.333em"></mspace></mrow><msub><mo>∑</mo><mrow><mi>b</mi><mo>∈</mo><msup><mi>ℬ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup></mrow></msub><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>b</mi></mrow></msub><mo>=</mo><mn>1</mn></mtd></mtr><mtr><mtd columnalign="left" style="text-align: left"><mn>𝟎</mn></mtd><mtd columnalign="left" style="text-align: left"><mtext mathvariant="normal">otherwise</mtext></mtd></mtr></mtable></mrow></mrow><annotation encoding="application/x-tex">\mathbf z^{(r)}_m | \mathbf z_m, \mathbf {\mathscr B}^{(r)} = \begin{cases} \mathbf z^{(r)}_m &amp; \text{if } \sum_{b \in \mathbf{\mathscr B}^{(r)}} z_{m,b}  = 1\\ \mathbf 0 &amp; \text{otherwise} \end{cases}</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐳</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf z_m</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><annotation encoding="application/x-tex">\mathbf z^{(r)}_m</annotation></semantics></math>
are vectors of indicators
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0</mn><annotation encoding="application/x-tex">0</annotation></semantics></math>
or
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>1</mn><annotation encoding="application/x-tex">1</annotation></semantics></math>)
identifying the broad-scale and regional populations that individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belongs to.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>ℬ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf{\mathscr B}^{(r)}</annotation></semantics></math>
denotes the broad-scale populations that belong to the areas represented
by the reporting groups of region
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>r</mi><annotation encoding="application/x-tex">r</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>𝟎</mn><annotation encoding="application/x-tex">\mathbf 0</annotation></semantics></math>
is a vector of all zeros.</p>
<p>Ultimately, we want to estimate the fraction of individuals in the
mixture that come from each of the sub-regional populations, as well as
from any of the populations in the broad-scale baseline that are not
associated with a regional baseline.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>p</mi><mi>k</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><annotation encoding="application/x-tex">p^{(r)}_k</annotation></semantics></math>
denotes the mixture proportion of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math><sup>th</sup>
population in region
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>r</mi><annotation encoding="application/x-tex">r</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mi>b</mi></msub><annotation encoding="application/x-tex">p_b</annotation></semantics></math>
denotes the mixture proportion of population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>b</mi><annotation encoding="application/x-tex">b</annotation></semantics></math>
in the broad-scale baseline. Thus, we endeavor to estimate the mixture
proportions
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>p</mi><mi>k</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><annotation encoding="application/x-tex">p^{(r)}_k</annotation></semantics></math>
for each
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo>,</mo><mi>k</mi><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(r, k)</annotation></semantics></math>
such that
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>r</mi><mo>=</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>R</mi></mrow><annotation encoding="application/x-tex">r = 1, 2, ..., R</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mo>=</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>K</mi><mi>r</mi></msub></mrow><annotation encoding="application/x-tex">k = 1, 2, ..., K_r</annotation></semantics></math>
along with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mi>b</mi></msub><annotation encoding="application/x-tex">p_b</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>b</mi><mo>∈</mo><msub><mi>ℬ</mi><mo mathvariant="bold">*</mo></msub></mrow><annotation encoding="application/x-tex">b \in \mathbf{\mathscr B_{*}}</annotation></semantics></math>,
with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ℬ</mi><mo mathvariant="bold">*</mo></msub><annotation encoding="application/x-tex">\mathbf{\mathscr B_{*}}</annotation></semantics></math>
denoting the broad-scale populations that do not belong to any areas
represented by regional baselines. Lastly, we multiply
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf {p}^{(r)}</annotation></semantics></math>
by
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mo>∑</mo><mrow><mi>b</mi><mo>∈</mo><msup><mi>ℬ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup></mrow></msub><msub><mi>p</mi><mi>b</mi></msub></mrow><annotation encoding="application/x-tex">\sum_{b \in \mathbf{\mathscr B}^{(r)}} p_b</annotation></semantics></math>
for each region
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>r</mi><annotation encoding="application/x-tex">r</annotation></semantics></math>,
so the scaled
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf{p}^{(r)}</annotation></semantics></math>
for all regions and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mi>b</mi></msub><annotation encoding="application/x-tex">p_b</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>b</mi><mo>∈</mo><msub><mi>ℬ</mi><mo mathvariant="bold">*</mo></msub></mrow><annotation encoding="application/x-tex">b \in \mathbf{\mathscr B_{*}}</annotation></semantics></math>
would sum to one.</p>
</div>
<div class="section level3">
<h3 id="gibbs-sampler-where-the-fun-go-round-and-round">Gibbs Sampler: where the fun go round and round<a class="anchor" aria-label="anchor" href="#gibbs-sampler-where-the-fun-go-round-and-round"></a>
</h3>
<p>Deriving the values of parameters in each stage of the integrated
multistage model requires finding the joint posterior distribution of
Pella-Masuda model in each stage,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝐩</mi><mo>,</mo><mi>𝐪</mi><mo>,</mo><mi>𝐳</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><mi>𝛂</mi><mo>,</mo><mi>𝛃</mi></mrow><annotation encoding="application/x-tex">\mathbf p, \mathbf q,\mathbf z, \mathbf y, \mathbf\alpha, \mathbf\beta</annotation></semantics></math>.
In this section, we will introduce the concepts and algorithm to sample
from this posterior distribution in a single baseline Pella-Masuda
model, which then can be extend to an integrated multistage
framework.</p>
<p>Gibbs sampler is a type of MCMC methods that sequentially sample
parameter values from a Markov chain. With enough sampling, the Markov
chain will eventually converge to the desire distribution of interest.
The most appealing quality of Gibbs sampler is its reduction of a
multivariate problem (such as Pella-Masuda model) to a series of more
manageable lower-dimensional problems. A full description of Gibbs
sampler and MCMC methods is beyond the scope of this document; however,
further information can be found in numerous resources devoting to
Bayesian data analysis (see Carlin &amp; Louis 2009; Robert &amp;
Casella 2010; Gelman et al. 2014)</p>
<p>To illustrate, suppose we would like to determine the joint posterior
distribution of interest,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐲</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf \theta|\mathbf y)</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝛉</mi><mo>=</mo><mo stretchy="false" form="prefix">{</mo><msub><mi>θ</mi><mn>1</mn></msub><mo>,</mo><msub><mi>θ</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>θ</mi><mi>K</mi></msub><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">\mathbf \theta = \{\theta_1, \theta_2,..., \theta_K\}</annotation></semantics></math>.
Most likely the multivariate
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐲</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf \theta|\mathbf y)</annotation></semantics></math>
would be too complicated to sample from. However, if we can figure out
how to break up the joint posterior distribution into individual full
conditional distributions, each parameter in
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝛉</mi><annotation encoding="application/x-tex">\mathbf \theta</annotation></semantics></math>
can be sampled one by one sequentially using a Gibbs sampler algorithm.
The process begins with an arbitrary set of starting values
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>θ</mi><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><msubsup><mi>θ</mi><mn>3</mn><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msubsup><mi>θ</mi><mi>K</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup></mrow><annotation encoding="application/x-tex">\theta^{(0)}_2, \theta^{(0)}_3,..., \theta^{(0)}_K</annotation></semantics></math>
and proceeds as follows:</p>
<p>For
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>t</mi><mo>=</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>T</mi></mrow><annotation encoding="application/x-tex">t = 1,2,...,T</annotation></semantics></math>,
repeat</p>
<ol style="list-style-type: decimal">
<li><p>Draw
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>θ</mi><mn>1</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><annotation encoding="application/x-tex">\theta^{(t)}_1</annotation></semantics></math>
from
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>θ</mi><mn>1</mn></msub><mo stretchy="false" form="prefix">|</mo><msubsup><mi>θ</mi><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><msubsup><mi>θ</mi><mn>3</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msubsup><mi>θ</mi><mi>k</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>𝐲</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\theta_1|\theta^{(t-1)}_2, \theta^{(t-1)}_3,..., \theta^{(t-1)}_k, \mathbf y)</annotation></semantics></math></p></li>
<li>
<p>Draw
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>θ</mi><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><annotation encoding="application/x-tex">\theta^{(t)}_2</annotation></semantics></math>
from
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>θ</mi><mn>2</mn></msub><mo stretchy="false" form="prefix">|</mo><msubsup><mi>θ</mi><mn>1</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><msubsup><mi>θ</mi><mn>3</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msubsup><mi>θ</mi><mi>k</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>𝐲</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\theta_2|\theta^{(t)}_1, \theta^{(t-1)}_3,..., \theta^{(t-1)}_k, \mathbf y)</annotation></semantics></math></p>
<p>⋮</p>
</li>
</ol>
<!-- --><ol start="11" style="list-style-type: lower-alpha">
<li>Draw
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>θ</mi><mi>k</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><annotation encoding="application/x-tex">\theta^{(t)}_k</annotation></semantics></math>
from
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>θ</mi><mi>k</mi></msub><mo stretchy="false" form="prefix">|</mo><msubsup><mi>θ</mi><mn>1</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><msubsup><mi>θ</mi><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msubsup><mi>θ</mi><mrow><mi>k</mi><mo>−</mo><mn>1</mn></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>𝐲</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\theta_k|\theta^{(t)}_1, \theta^{(t)}_2,..., \theta^{(t)}_{k-1}, \mathbf y)</annotation></semantics></math>
</li>
</ol>
<p>This would work best if the full conditionals are some known
distributions that we can easily sample from (although it’s not
required). In our case with Pella-Masuda model, we rely on two main
concepts, the Bayes theorem and conjugacy, to do the trick. Briefly, for
estimating parameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝛉</mi><annotation encoding="application/x-tex">\mathbf\theta</annotation></semantics></math>
from data
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐃</mi><annotation encoding="application/x-tex">\mathbf D</annotation></semantics></math>,
according to Bayes Rule,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mfrac><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac></mrow><annotation encoding="application/x-tex">p(\mathbf\theta|\mathbf D) = \displaystyle \frac{p(\mathbf D|\mathbf\theta)p(\mathbf\theta)}{p(\mathbf D)}</annotation></semantics></math>.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf\theta|\mathbf D)</annotation></semantics></math>
is the joint posterior distribution for parameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝛉</mi><annotation encoding="application/x-tex">\mathbf\theta</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf D|\mathbf\theta)</annotation></semantics></math>
is the likelihood of observing the data given the parameters,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf\theta)</annotation></semantics></math>
is the prior distribution of the parameters, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf D)</annotation></semantics></math>
is the constant marginal distribution of the data.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf D)</annotation></semantics></math>
is often mathematically difficult to obtain; however, because
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf D)</annotation></semantics></math>
is a constant number, we can ignore it by reducing the posterior
distribution to
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf\theta|\mathbf D) \propto p(\mathbf D|\mathbf\theta)p(\mathbf\theta)</annotation></semantics></math>.</p>
<p>So, how does Bayes Rule help us estimating parameters in the
Pella-Masuda model? First, the joint posterior distribution has to be
split up into smaller pieces. That is, we separate the joint posterior
into likelihood of the data and priors for the parameters:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo>,</mo><mi>𝐪</mi><mo>,</mo><mi>𝐳</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><mi>𝛂</mi><mo>,</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf p, \mathbf q, \mathbf z, \mathbf y, \mathbf\alpha, \mathbf\beta)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\propto p(\mathbf x|\mathbf z, \mathbf q) p(\mathbf y|\mathbf q) \cdot p(\mathbf p|\mathbf\alpha) p(\mathbf q|\mathbf\beta) p(\mathbf z|\mathbf p)</annotation></semantics></math></p>
<p>With some re-arrangements and hand-waving, we arrive at the full
conditional distributions for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐩</mi><annotation encoding="application/x-tex">\mathbf p</annotation></semantics></math>:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf x|\mathbf z, \mathbf q) p(\mathbf y|\mathbf q) \cdot p(\mathbf p|\mathbf\alpha) p(\mathbf q|\mathbf\beta) p(\mathbf z|\mathbf p)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>=</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">= p(\mathbf x|\mathbf z, \mathbf q) p(\mathbf y|\mathbf q) p(\mathbf q|\mathbf\beta) \cdot p(\mathbf z|\mathbf p) p(\mathbf p|\mathbf\alpha)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\propto p(\mathbf x,\mathbf y,\mathbf z|\mathbf q) p(\mathbf q|\mathbf\beta) \cdot p(\mathbf z|\mathbf p) p(\mathbf p|\mathbf\alpha)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><mi>𝐳</mi><mo>,</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\propto p(\mathbf q|\mathbf x,\mathbf y,\mathbf z,\mathbf\beta) \cdot p(\mathbf p|\mathbf z,\mathbf\alpha)</annotation></semantics></math></p>
<p>Next, we take advantage of a mathematical property called
<em>conjugacy</em> to help us determine the conditional distributions.
Based on this property, the posterior distribution follows the same
parametric form as the prior distribution when prior is a <em>conjugate
family</em> for the likelihood. For example, if the likelihood of data
is binomial distribution and the prior of parameter is beta
distribution, then the posterior is also beta distribution because beta
is a conjugate family for binomial. There are many conjugate families,
and Dirichlet and multinomial are another example.</p>
<p>Utilizing conjugacy property, we will determine each of the full
conditional distributions for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐩</mi><annotation encoding="application/x-tex">\mathbf p</annotation></semantics></math>.</p>
<div class="section level4">
<h4 id="conditional-distribution-pqx-y-z-beta">Conditional distribution p(q|x, y, z,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>β</mi><annotation encoding="application/x-tex">\beta</annotation></semantics></math>)<a class="anchor" aria-label="anchor" href="#conditional-distribution-pqx-y-z-beta"></a>
</h4>
<p>We determine that
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><mi>𝐳</mi><mo>,</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf q|\mathbf x, \mathbf y, \mathbf z, \mathbf \beta)</annotation></semantics></math>
is Dirichlet-distributed because Dirichlet prior
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf q|\mathbf \beta)</annotation></semantics></math>
is a conjugate family for the multinomial likelihoods
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf x|\mathbf z, \mathbf q)</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf y|\mathbf q)</annotation></semantics></math>.
To determine the exact parameterization for the posterior distribution,
we need to derive the prior and likelihoods first.</p>
<p>Likelihood
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf x|\mathbf z, \mathbf q)</annotation></semantics></math>
can be derived in two steps. The first step we conditioned the
likelihood on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐳</mi><annotation encoding="application/x-tex">\mathbf z</annotation></semantics></math>
so that</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msup><mrow><mo stretchy="true" form="prefix">[</mo><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub></msup></mrow><annotation encoding="application/x-tex">p(\mathbf x|\mathbf z, \mathbf q) \propto \displaystyle \prod^{M}_{m=1} \prod^{K}_{k=1} [f(\mathbf x_m|\mathbf q_k)]^{z_{m,k}}</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">f(\mathbf x_m|\mathbf q_k)</annotation></semantics></math>
is the relative frequency of multi-locus genotype for individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
in population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>.
In the next step, we derive
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">f(\mathbf x_m|\mathbf q_k)</annotation></semantics></math>:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub></msubsup></mrow><annotation encoding="application/x-tex">f(\mathbf x_m|\mathbf q_k) \propto \displaystyle \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{x_{m,l,j}}_{k,l,j}</annotation></semantics></math></p>
<p>Then we combine the two,</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msup><mrow><mo stretchy="true" form="prefix">[</mo><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub></msup></mrow><annotation encoding="application/x-tex">p(\mathbf x|\mathbf z, \mathbf q) \propto \displaystyle \prod^{M}_{m=1} \prod^{K}_{k=1} [f(\mathbf x_m|\mathbf q_k)]^{z_{m,k}}</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><mrow><mo stretchy="true" form="prefix">[</mo><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>⋅</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub></mrow></msubsup><mo stretchy="true" form="postfix">]</mo></mrow></mrow><annotation encoding="application/x-tex">\propto \displaystyle \prod^{M}_{m=1} \prod^{K}_{k=1} [\displaystyle \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{x_{m,l,j} \cdot z_{m,k}}_{k,l,j}]</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>⋅</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow></msubsup></mrow><annotation encoding="application/x-tex">\propto \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\sum^{M}_{m=1} (x_{m,l,j} \cdot z_{m,k})}_{k,l,j}</annotation></semantics></math></p>
<p>Deriving likelihood
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf y|\mathbf q)</annotation></semantics></math>
is more straightforward. It is the product of relative frequency of
multi-locus genotype for each population:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub></msubsup></mrow><annotation encoding="application/x-tex">p(\mathbf y|\mathbf q) \propto \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{y_{k,l,j}}_{k,l,j}</annotation></semantics></math></p>
<p>And
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>q</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(q|\mathbf \beta)</annotation></semantics></math>
is Dirichlet prior distribution. Its probability density has a kernel<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;Without involving too much math, &lt;em&gt;kernel&lt;/em&gt; here
refers to what is left over after factoring out the constant from a
probability density function (PDF). For example, The PDF of Dirichlet
distribution is
&lt;math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mfrac&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;mrow&gt;&lt;mi&gt;B&lt;/mi&gt;&lt;mrow&gt;&lt;mo stretchy="true" form="prefix"&gt;(&lt;/mo&gt;&lt;mi&gt;𝛂&lt;/mi&gt;&lt;mo stretchy="true" form="postfix"&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/mrow&gt;&lt;/mfrac&gt;&lt;msubsup&gt;&lt;mo&gt;∏&lt;/mo&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mo&gt;=&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mrow&gt;&lt;mi&gt;K&lt;/mi&gt;&lt;/msubsup&gt;&lt;msubsup&gt;&lt;mi&gt;x&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mrow&gt;&lt;msub&gt;&lt;mi&gt;α&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/msub&gt;&lt;mo&gt;−&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mrow&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;annotation encoding="application/x-tex"&gt;\frac{1}{B(\mathbf \alpha)} \displaystyle \prod^K_{i=1} x^{\alpha_i - 1}_i&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;.
Beta function
&lt;math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mi&gt;B&lt;/mi&gt;&lt;mrow&gt;&lt;mo stretchy="true" form="prefix"&gt;(&lt;/mo&gt;&lt;mi&gt;𝛂&lt;/mi&gt;&lt;mo stretchy="true" form="postfix"&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/mrow&gt;&lt;annotation encoding="application/x-tex"&gt;B(\mathbf \alpha)&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;
can be factored out, and
&lt;math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msubsup&gt;&lt;mo&gt;∏&lt;/mo&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mo&gt;=&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mrow&gt;&lt;mi&gt;K&lt;/mi&gt;&lt;/msubsup&gt;&lt;msubsup&gt;&lt;mi&gt;x&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mrow&gt;&lt;msub&gt;&lt;mi&gt;α&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/msub&gt;&lt;mo&gt;−&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mrow&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;annotation encoding="application/x-tex"&gt;\displaystyle \prod^K_{i=1} x^{\alpha_i - 1}_i&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;
is the kernel.&lt;/p&gt;'><sup>6</sup></a> of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐪</mi><mrow><mi>𝛃</mi><mo>−</mo><mn>1</mn></mrow></msup><annotation encoding="application/x-tex">\mathbf q^{\mathbf \beta - 1}</annotation></semantics></math>.
We can express the likelihood as</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msub><mi>β</mi><mrow><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">p(\mathbf q|\mathbf \beta) \propto \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\beta_{l,j} - 1}_{k,l,j}</annotation></semantics></math>.</p>
<p>Put all the likelihoods together,</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><mi>𝐳</mi><mo>,</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf q|\mathbf x,\mathbf y,\mathbf z,\mathbf\beta) \propto p(\mathbf x|\mathbf z, \mathbf q) p(\mathbf y|\mathbf q) p(\mathbf q|\mathbf\beta)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>⋅</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow></msubsup><mo>⋅</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub></msubsup><mo>⋅</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msub><mi>β</mi><mrow><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">\propto \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\sum^{M}_{m=1} (x_{m,l,j} \cdot z_{m,k})}_{k,l,j} \cdot \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{y_{k,l,j}}_{k,l,j} \cdot \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\beta_{l,j} - 1}_{k,l,j}</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>=</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>⋅</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>+</mo><msub><mi>β</mi><mrow><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">= \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\sum^{M}_{m=1} (x_{m,l,j} \cdot z_{m,k}) + y_{k,l,j} + \beta_{l,j} - 1}_{k,l,j}</annotation></semantics></math></p>
<p>It is <em>elementary</em> to recognize that
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>⋅</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>+</mo><msub><mi>β</mi><mrow><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">\displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\sum^{M}_{m=1} (x_{m,l,j} \cdot z_{m,k}) + y_{k,l,j} + \beta_{l,j} - 1}_{k,l,j}</annotation></semantics></math>
is the kernel for Dirichlet distribution. Hence,</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐪</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi></mrow></msub><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><mi>𝐳</mi><mo>,</mo><mi>𝛃</mi><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>+</mo><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>+</mo><msub><mi>β</mi><mrow><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf q_{k,l}|\mathbf x,\mathbf y,\mathbf z,\mathbf\beta \sim Dirich(\displaystyle \sum^{M}_{m=1} x_{m,l,j} z_{m,k} + y_{k,l,j} + \beta_{l,j})</annotation></semantics></math>.</p>
</div>
<div class="section level4">
<h4 id="conditional-distribution-ppz-alpha">Conditional distribution p(p|z,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>)<a class="anchor" aria-label="anchor" href="#conditional-distribution-ppz-alpha"></a>
</h4>
<p>Using the same logic as previously,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf p|\mathbf z,\mathbf\alpha)</annotation></semantics></math>
is also Dirichlet-distributed due to a Dirichlet prior
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf p|\mathbf\alpha)</annotation></semantics></math>
and a multinomial likelihood
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf z|\mathbf p)</annotation></semantics></math>.</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf p|\mathbf z,\mathbf\alpha) \propto p(\mathbf z|\mathbf p) p(\mathbf p|\mathbf\alpha)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mi>p</mi><mi>k</mi><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub></msubsup><mo>⋅</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mi>p</mi><mi>k</mi><mrow><msub><mi>α</mi><mi>k</mi></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">\propto \displaystyle \prod^{M}_{m=1} \prod^{K}_{k=1} p^{z_{m,k}}_k \cdot \prod^{K}_{k=1} p^{\alpha_k - 1}_k</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mi>p</mi><mi>k</mi><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>+</mo><msub><mi>α</mi><mi>k</mi></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">\propto \displaystyle \prod^{K}_{k=1} p^{\sum^M_{m=1}z_{m,k} + \alpha_k - 1}_k</annotation></semantics></math></p>
<p>Once again, we recognize it as the kernel for Dirichlet
distribution:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛂</mi><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>+</mo><msub><mi>α</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf p|\mathbf z,\mathbf\alpha \sim Dirich(\displaystyle \sum^M_{m=1}z_{m,k} + \alpha_k)</annotation></semantics></math></p>
</div>
<div class="section level4">
<h4 id="algorithm">Algorithm<a class="anchor" aria-label="anchor" href="#algorithm"></a>
</h4>
<p>There is one more distribution to figure out before we can start our
Gibbs sampler routine (and you thought we’re all set, lol). We would
need to know how to sample
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐳</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo>,</mo><mi>𝐪</mi><mo>,</mo><msub><mi>𝐱</mi><mi>m</mi></msub></mrow><annotation encoding="application/x-tex">\mathbf z_m|\mathbf p, \mathbf q, \mathbf x_m</annotation></semantics></math>,
the population identity for individual fish
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
(in components 1 and 2) given the population proportions and genotype.
If the probability of fish
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belong to population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mi>k</mi></msub><annotation encoding="application/x-tex">p_k</annotation></semantics></math>,
and the likelihood of observing relative frequency of genotype for fish
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
in population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">f(\mathbf x_m|\mathbf q_k)</annotation></semantics></math>,
then the probability of fish
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belong to population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
given the population proportions and genotype is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mfrac><mrow><msub><mi>p</mi><mi>k</mi></msub><mo>⋅</mo><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>k</mi><mi>′</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msub><mi>p</mi><mrow><mi>k</mi><mi>′</mi></mrow></msub><mo>⋅</mo><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mrow><mi>k</mi><mi>′</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac><annotation encoding="application/x-tex">\displaystyle \frac{p_k \cdot f(\mathbf x_m|\mathbf q_k)}{\sum^K_{k'=1}p_{k'} \cdot f(\mathbf x_m|\mathbf q_{k'})}</annotation></semantics></math>.
The denominator should sum to one, so we only need to calculate the
numerator.</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐳</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo>,</mo><mi>𝐪</mi><mo>,</mo><msub><mi>𝐱</mi><mi>m</mi></msub></mrow><annotation encoding="application/x-tex">\mathbf z_m|\mathbf p, \mathbf q, \mathbf x_m</annotation></semantics></math>
has the following distribution:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐳</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo>,</mo><mi>𝐪</mi><mo>,</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><msub><mi>𝐰</mi><mi>m</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf z_m|\mathbf p, \mathbf q, \mathbf x_m \sim Mult(1, \mathbf{w}_m)</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>w</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>=</mo><msub><mi>p</mi><mi>k</mi></msub><mo>⋅</mo><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">w_{m,k} = p_k \cdot f(\mathbf x_m|\mathbf q_k)</annotation></semantics></math>.
We draw the initial values for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐪</mi><mi>k</mi></msub><annotation encoding="application/x-tex">\mathbf q_k</annotation></semantics></math>
based on its prior distribution.</p>
<p>Once we figured out all the pieces in the Gibbs sampler for the
single baseline framework, we can extend the concept to a multistage
framework. Conceptually, we use a Gibbs sampler to sample from the full
conditionals of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐩</mi><annotation encoding="application/x-tex">\mathbf p</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐳</mi><annotation encoding="application/x-tex">\mathbf z</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf q^{(r)}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf p^{(r)}</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf z^{(r)}</annotation></semantics></math>.
The full conditional distributions of the broad-scale baseline in the
multistage framework stay the same as their counterparts in a single
baseline framework. The full conditional distributions for the Gibbs
sampler at the regional stage are:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>𝐪</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐱</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝐲</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝛃</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msubsup><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><msubsup><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>+</mo><msubsup><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>+</mo><msubsup><mi>β</mi><mrow><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf q^{(r)}_{k,l} | \mathbf x^{(r)},\mathbf y^{(r)},\mathbf z^{(r)},\mathbf\beta^{(r)} \sim Dirich(\displaystyle \sum^{M}_{m=1} x^{(r)}_{m,l,j} z^{(r)}_{m,k} + y^{(r)}_{k,l,j} + \beta^{(r)}_{l,j})</annotation></semantics></math>,</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝐳</mi><mo>,</mo><msup><mi>ℬ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝛂</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>b</mi></mrow></msub><mo>,</mo><msubsup><mi>ℬ</mi><mrow><mi>m</mi><mo>,</mo><mi>b</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><msubsup><mi>α</mi><mi>k</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf p^{(r)} | \mathbf z^{(r)}, \mathbf z, \mathbf{\mathscr B}^{(r)}, \mathbf \alpha^{(r)} \sim Dirich(\displaystyle \sum^M_{m=1} (z^{(r)}_{m,k} | z_{m,b}, \mathscr B^{(r)}_{m,b}) + \alpha^{(r)}_k)</annotation></semantics></math>,</p>
<p>and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><msubsup><mi>𝐰</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf z^{(r)}_m | \mathbf p^{(r)}, \mathbf q^{(r)}, \mathbf x^{(r)}_m \sim Mult(1, \mathbf{w}^{(r)}_m)</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>w</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>∝</mo><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mi>p</mi><mi>k</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>⋅</mo><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><msup><mi>q</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub></msubsup><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">w^{(r)}_{m,k} \propto (p^{(r)}_k \cdot \displaystyle \prod^{L}_{l=1} \prod^{J_l}_{j=1} {q^{(r)}}^{x_{m,l,j}}_{k,l,j})</annotation></semantics></math>.</p>
<p>We initiate the Gibbs sampler with starting values for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf p^{(0)}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf q^{(0)}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix" mathvariant="bold">(</mo><mi>𝐫</mi><mo stretchy="true" form="postfix" mathvariant="bold">)</mo></mrow></msup><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf {p^{(r)}}^{(0)}</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix" mathvariant="bold">(</mo><mi>𝐫</mi><mo stretchy="true" form="postfix" mathvariant="bold">)</mo></mrow></msup><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf {q^{(r)}}^{(0)}</annotation></semantics></math>
based on their prior distributions. We use subscript
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">^{(t)}</annotation></semantics></math>
to denote
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>t</mi><annotation encoding="application/x-tex">t</annotation></semantics></math><sup>th</sup>
iteration of the Gibbs sampler. Sampling for the fully Bayesian model
proceeds as follows:</p>
<p>For
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>t</mi><mo>=</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>T</mi></mrow><annotation encoding="application/x-tex">t = 1,2,...,T</annotation></semantics></math>,
repeat</p>
<ol style="list-style-type: decimal">
<li><p>Determine the group memberships of mixture individuals at the
broad-scale stage,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><msub><mi>𝐰</mi><mi>m</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf z^{(t)}_m | \mathbf p^{(t-1)}, \mathbf q^{(t-1)}, \mathbf x_m \sim Mult(1, \mathbf{w}_m)</annotation></semantics></math>.</p></li>
<li><p>Determine the group memberships of mixture individuals for each
sub-region at the regional stage,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="false" form="prefix">|</mo><msup><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><msubsup><mi>𝐰</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">{\mathbf z^{(r)}_m}^{(t)} | {\mathbf p^{(r)}}^{(t-1)}, {\mathbf q^{(r)}}^{(t-1)}, \mathbf x_m^{(r)} \sim Mult(1, \mathbf{w}^{(r)}_m)</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>r</mi><mo>=</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>R</mi></mrow><annotation encoding="application/x-tex">r = 1, 2, ..., R</annotation></semantics></math>.</p></li>
<li><p>Draw updated values,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf q^{(t)}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf p^{(t)}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">{{\mathbf q}^{(r)}}^{(t)}</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">{{\mathbf p}^{(r)}}^{(t)}</annotation></semantics></math>
from
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf q | \mathbf x, \mathbf y, \mathbf z^{(t)}, \mathbf \beta)</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf p | \mathbf z^{(t)}, \mathbf \alpha)</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐱</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝐲</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝛃</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf q^{(r)} | \mathbf x^{(r)}, \mathbf y^{(r)}, {\mathbf z^{(r)}}^{(t)}, \mathbf \beta^{(r)})</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="false" form="prefix">|</mo><msup><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix" mathvariant="bold">(</mo><mi>𝐫</mi><mo stretchy="true" form="postfix" mathvariant="bold">)</mo></mrow></msup><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>ℬ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝛂</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf p^{(r)} | \mathbf {z^{(r)}}^{(t)}, \mathbf z^{(t)}, \mathbf{\mathscr B}^{(r)}, \mathbf \alpha^{(r)})</annotation></semantics></math>
respectively.</p></li>
</ol>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>T</mi><annotation encoding="application/x-tex">T</annotation></semantics></math>
should be large enough to ensure the simulations converge to the
posterior distribution. Usually it takes thousands of iterations.
Implementing the conditional GSI model only requires a slight
modification from the above algorithm.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>r</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf q^{(r)}</annotation></semantics></math>
only need to be calculated once at the initial step without further
updates, otherwise the procedures remain the same.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Brooks, S. P., and A. Gelman. 1998. General methods for monitoring
convergence of iterative simulations. <em>Journal of Computational and
Graphical Statistics</em>. 7:434–455.</p>
<p>Carlin, B. and T. Louis. 2009. <em>Bayesian Methods for Data
Analysis, 3rd Edition</em>. CRC Press. New York.</p>
<p>Gelman, A., and D. B. Rubin. 1992. Inference from iterative
simulation using multiple sequences. <em>Statistical Science</em>.
7:457–472.</p>
<p>Gelman, A., J. Carlin, H. Stern, D. Dunson, A. Vehtari and D. Rubin.
2014. <em>Bayesian Data Analysis, 3rd Edition</em>. CRC Press. New
York.</p>
<p>Johnson, N.L., Kotz, S., and Balakrishnan, N. 1997. Discrete
multivariate distributions. Wiley &amp; Sons, New York.</p>
<p>Lee. E., T. Dann, and H. Hoyt. 2021. Yukon River Chinook Genetic
Baseline Improvements. Yukon River Panel Restoration and Enhancement
Fund Final Report, URE-163-19N.</p>
<p>Moran, B.M. and E.C. Anderson. 2018. Bayesian inference from the
conditional genetic stock identification model. <em>Canadian Journal of
Fisheries and Aquatic Sciences</em>. 76(4):551-560. <a href="https://doi.org/10.1139/cjfas-2018-0016" class="external-link uri">https://doi.org/10.1139/cjfas-2018-0016</a></p>
<p>Pella, J. and M. Masuda. 2001. Bayesian methods for analysis of stock
mixtures from genetic characters. <em>Fish. Bull.</em> 99:151–167.</p>
<p>Robert, C. and G. Casella. 2010. <em>Introducing Monte Carlo Methods
with R</em>. Springer. New York.</p>
<p>Templin, W. D., J. E. Seeb, J. R. Jasper, A. W. Barclay, L. W. Seeb.
2011. Genetic differentiation of Alaska Chinook salmon: the missing link
for migratory studies. <em>Mol Ecol Resour</em>. 11(Suppl 1):226-246. <a href="doi:10.1111/j.1755-0998.2010.02968.x" class="uri">doi:10.1111/j.1755-0998.2010.02968.x</a>.</p>
</div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bobby Hsu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
